---
title: "Effect_size_2lms_cor"
author: "Lauren Blake"
date: "November 29, 2017"
output: html_document
---

## Prepare data

```{r}
# import sample labels
samples <- read.delim("../data/Sample_info_RNAseq_RIN.txt")
# expression
exprs <- read.table("../data/human_chimp_orth_exp_methyl_7725_hum.txt", sep="")
# methylation data
methyl <- read.csv("../data/chimp_human_orth_7725_avg_methyl_per_ts_gene.txt", sep="")
# import previous computed results
df_tmp <- get(load("../Limma_output_fit1.rda"))

# <---- expression data matching/cleaning
exprs_tmp <- exprs[,2:48]
exprs_tmp <- exprs_tmp[rownames(exprs_tmp) %in% rownames(df_tmp),]  
exprs_subset <- exprs_tmp[,1:31]
colnames(exprs_subset) <- gsub(".x", "", colnames(exprs_subset))

# <---- methylation data matching/cleaning
methyl_tmp <- methyl[,-17]
methyl_tmp[,32] <- as.character(methyl_tmp[,32])
methyl_tmp <-  methyl_tmp[methyl_tmp[,32] %in% rownames(exprs_subset),]
methyl_subset <- methyl_tmp[,1:31]
rownames(methyl_subset) <- methyl_tmp[,32]


# check agreement between methylation data and expression data
all.equal(rownames(methyl_subset), rownames(exprs_subset))
all.equal(colnames(methyl_subset), colnames(exprs_subset))

# <---- sample data matching
samples_sub <- samples[samples$Sample_ID %in% colnames(exprs_subset),]
```

## Subsetting data to human samples heart vs. kidney pair. Running linear models on exprs ~ tissue + RIN and also on exprs|methylation ~ tissue + RIN

```{r}
# <---- subset data to the tissue pair
# compare human heart and kidney
index_sample <- which(samples_sub$Species == "human"& (samples_sub$Tissue == "heart"|samples_sub$Tissue == "kidney"))
exprs_pair <- exprs_subset[,index_sample]
methyl_pair <- methyl_subset[,index_sample]
species <- droplevels.factor(samples$Species[index_sample])
tissue <- droplevels.factor(samples$Tissue[index_sample])
RIN <- samples$RIN[index_sample]

# Extract parameters
N <- ncol(exprs_pair)
ngenes <- nrow(methyl_pair)
# <---- Model 1: exprs ~ tissue
# specify tissue coding
design_1 <- model.matrix(~tissue + RIN)
design_1[design_1[,2]==0,2] <- -1
model_1 <- lmFit(exprs_pair, design_1)


# <---- Model 3: exprs corrected for methylation ~ tissue
# specify design matrix
resid_exprs <- array(0, dim = dim(exprs_pair))
for (index in 1:nrow(exprs_pair)){
  resid_exprs[index,] <- lm(t(exprs_pair[index, ]) ~ t(methyl_pair[index, ]))$resid
}
rownames(resid_exprs) <- rownames(exprs_pair)
model_3 <- lmFit(resid_exprs, design_1)  

```

## Effect size differences  

```{r}
# get effect sizes
beta1 <- coef(model_1[,2])
beta3 <- coef(model_3[,2])
se_beta1 <- se_beta2 <- se_beta3 <- cov_beta13 <- vector("numeric", ngenes)
# <---- get variances of beta1^S tissue effect
se_beta1 <- model_1$sigma*sqrt((solve(t(design_1)%*%design_1))[2,2])
head(cbind(se_beta1,model_1$sigma*sqrt(model_1$cov.coefficients[2,2])))
# <---- get variances of beta2 methylation effect
for (g in 1:length(se_beta2)) {
  design_2g <- model.matrix(~ as.numeric(methyl_pair[g,]))  
  sigma_g <- model_1$sigma[g]
  se_beta2[index] <- sigma_g*sqrt((solve(t(design_2g)%*%design_2g))[2,2])
}
# <---- get variances of beta3 tissue effect
A <- solve(t(design_1)%*%design_1)%*%t(design_1)
contr.vector <- array(c(0,1), dim=c(2,1))
# compute beta3 by hand
for (g in 1:length(se_beta3)) {
  M_g <- t(methyl_pair[g,])
  design_2g <- model.matrix(~ as.numeric(M_g))  
  A_2g <- solve(t(design_2g)%*%design_2g)%*%t(design_2g)
  sigma_g <- model_1$sigma[g]
  var_beta2g <- (se_beta2^2)[g]
  var_part1 <- (se_beta1[g])^2
  var_part2 <- ( A%*%M_g%*%var_beta2g%*%t(M_g)%*%t(A) )[2,2]
  var_part3 <- ( 2*(sigma_g^2)*A%*%t(A_2g)%*%contr.vector%*%t(M_g)%*%t(A) )[2,2]
  se_beta3[g] <- sqrt(var_part1 + var_part2 + var_part3)
  }
# cov(beta1,beta3)
for (g in 1:length(cov_beta13)) {
  M_g <- t(methyl_pair[g,])
  design_2g <- model.matrix(~ as.numeric(M_g))  
  A_2g <- solve(t(design_2g)%*%design_2g)%*%t(design_2g)
  sigma_g <- model_1$sigma[g]
  var_part1 <- (se_beta1[g])^2
  cov_beta13[g] <- var_part1 - (sigma_g^2)*((A %*% t(A_2g) %*%contr.vector%*%t(M_g)%*%t(A))[2,2])
}
  
# cov(beta1-beta3)
cov_diff_sqrt <- sqrt(se_beta1^2 + se_beta3^2 - 2*cov_beta13)
beta_diff <- beta1-beta3

summary(se_beta1^2)
summary(se_beta3^2)
summary(cov_beta13)
summary(cov_diff_sqrt^2)


#col_cov_diff_sqrt <- cov_diff_sqrt/max(cov_diff_sqrt)
plot(x=beta1-beta3, y=cov_diff_sqrt,
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size", 
     main = "Difference of effect size (between 2 linear models)",
     pch=16, cex=.6)

# What is the summary of beta1 and beta3?

plot(beta1, beta3, xlab = "Effect size from lm 1", ylab = "Effect size from lm 2", main = "Effect size comparison of difference in human heart and human kidney")

summary(beta1-beta3)

# How genes have beta1 > beta3?

summary(beta1 > beta3)

# How genes have beta3 > beta1?

summary(beta1 < beta3)


```


Appears that the differences between the effect sizes center at approximately 0 log2 fold change. Also note that standard errors are large for when the effect sizes from before correcting are way smaller than after correcting.


## Run ASH

```{r}
library(ashr)
# mode = a number that corresponds to when there's no difference between 
# the effect sizes. I try for 0, because that seems to be the average here.
# try to experiment different modes and see what happens
fit_ash <- ash(as.vector(beta_diff), cov_diff_sqrt, mode=0)
names(fit_ash$result)
# lfsr is the minimum of PositiveProb and NegativeProb
# look at genes at various cut-off, say svalue < .01 etc. do they make sense?
# ind_sig <- which(fit_ash$result$svalue < .01)
head(fit_ash$result)
#ind_sig <- which(fit_ash$result$svalue < .01)
ind_sig <- (fit_ash$result$svalue < .01)
ind_sig[ind_sig == TRUE] <- "red"
ind_sig[ind_sig == FALSE] <- "black"
ind_sig <- as.data.frame(ind_sig)

make_plot <- cbind(beta1-beta3, cov_diff_sqrt, ind_sig, fit_ash$result$lfdr)

plot(x=make_plot[,1], y=make_plot[,2],
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Testing null hypothesis that the difference in effect sizes = 0",
     pch=16, cex=.6, col = make_plot[,3])
legend("topleft", legend=c("s < 0.01", "s >= 0.01"), col=c("red", "black"), pch=16:16, cex=0.8)

nrow(as.data.frame(ind_sig))






fit_ash <- ash(as.vector(beta_diff), cov_diff_sqrt, mode=1)
names(fit_ash$result)
# lfsr is the minimum of PositiveProb and NegativeProb
# look at genes at various cut-off, say svalue < .01 etc. do they make sense?
# ind_sig <- which(fit_ash$result$svalue < .01)
head(fit_ash$result)
#ind_sig <- which(fit_ash$result$svalue < .01)
ind_sig <- (fit_ash$result$svalue < .01)
ind_sig[ind_sig == TRUE] <- "red"
ind_sig[ind_sig == FALSE] <- "black"
ind_sig <- as.data.frame(ind_sig)


make_plot <- cbind(beta1-beta3, cov_diff_sqrt, ind_sig)

plot(x=make_plot[,1], y=make_plot[,2],
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Difference of effect size (human heart vs human kidney)",
     pch=16, cex=.6, col = make_plot[,3])


fit_ash <- ash(as.vector(beta_diff), cov_diff_sqrt, mode=-1)
names(fit_ash$result)
# lfsr is the minimum of PositiveProb and NegativeProb
# look at genes at various cut-off, say svalue < .01 etc. do they make sense?
# ind_sig <- which(fit_ash$result$svalue < .01)
head(fit_ash$result)
#ind_sig <- which(fit_ash$result$svalue < .01)
ind_sig <- (fit_ash$result$svalue < .01)
ind_sig[ind_sig == TRUE] <- "red"
ind_sig[ind_sig == FALSE] <- "black"
ind_sig <- as.data.frame(ind_sig)


make_plot <- cbind(beta1-beta3, cov_diff_sqrt, ind_sig)

plot(x=make_plot[,1], y=make_plot[,2],
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Difference of effect size (human heart vs human kidney)",
     pch=16, cex=.6, col = make_plot[,3])




fit_ash <- ash(as.vector(beta_diff), cov_diff_sqrt, mode=2)
names(fit_ash$result)
# lfsr is the minimum of PositiveProb and NegativeProb
# look at genes at various cut-off, say svalue < .01 etc. do they make sense?
# ind_sig <- which(fit_ash$result$svalue < .01)
head(fit_ash$result)
#ind_sig <- which(fit_ash$result$svalue < .01)
ind_sig <- (fit_ash$result$svalue < .01)
ind_sig[ind_sig == TRUE] <- "red"
ind_sig[ind_sig == FALSE] <- "black"
ind_sig <- as.data.frame(ind_sig)


make_plot <- cbind(beta1-beta3, cov_diff_sqrt, ind_sig)

plot(x=make_plot[,1], y=make_plot[,2],
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Difference of effect size (human heart vs human kidney)",
     pch=16, cex=.6, col = make_plot[,3])



```


A commonly used diagnostics in checking ash results is to compare the raw effect sizes versus the posterior mean estimated by ash. This gives you an idea how much shrinkage that was done.
Results here: Observe shrinkage of effect size for positive effect size but not for negative.

```{r}
plot(x=fit_ash$result$betahat,
     y=fit_ash$result$PosteriorMean,
     xlab = "Raw difference", ylab = "Posterior estimate", 
     main = "Difference in effect size before and after ASH",
     pch =16,cex=.6)
abline(0,1,col="red")
```




```{r}
# import sample labels
samples <- read.delim("../data/Sample_info_RNAseq_RIN.txt")
# expression
exprs <- read.table("../data/human_chimp_orth_exp_methyl_7725_hum.txt", sep="")
# methylation data
methyl <- read.csv("../data/chimp_human_orth_7725_avg_methyl_per_ts_gene.txt", sep="")
# import previous computed results
df_tmp <- get(load("../Limma_output_fit1.rda"))

# <---- expression data matching/cleaning
exprs_tmp <- exprs[,2:48]
exprs_tmp <- exprs_tmp[(rownames(exprs_tmp) %in% rownames(df_tmp) == FALSE),]  
exprs_subset <- exprs_tmp[,1:31]
colnames(exprs_subset) <- gsub(".x", "", colnames(exprs_subset))

# <---- methylation data matching/cleaning
methyl_tmp <- methyl[,-17]
methyl_tmp[,32] <- as.character(methyl_tmp[,32])
methyl_tmp <-  methyl_tmp[methyl_tmp[,32] %in% rownames(exprs_subset),]
methyl_subset <- methyl_tmp[,1:31]
rownames(methyl_subset) <- methyl_tmp[,32]


# check agreement between methylation data and expression data
all.equal(rownames(methyl_subset), rownames(exprs_subset))
all.equal(colnames(methyl_subset), colnames(exprs_subset))

# <---- sample data matching
samples_sub <- samples[samples$Sample_ID %in% colnames(exprs_subset),]
```

## Subsetting data to human samples heart vs. kidney pair. Running linear models on exprs ~ tissue + RIN and also on exprs|methylation ~ tissue + RIN

```{r}
# <---- subset data to the tissue pair
# compare human heart and kidney
index_sample <- which(samples_sub$Species == "human"& (samples_sub$Tissue == "heart"|samples_sub$Tissue == "kidney"))
exprs_pair <- exprs_subset[,index_sample]
methyl_pair <- methyl_subset[,index_sample]
species <- droplevels.factor(samples$Species[index_sample])
tissue <- droplevels.factor(samples$Tissue[index_sample])
RIN <- samples$RIN[index_sample]

# Extract parameters
N <- ncol(exprs_pair)
ngenes <- nrow(methyl_pair)
# <---- Model 1: exprs ~ tissue
# specify tissue coding
design_1 <- model.matrix(~tissue + RIN)
design_1[design_1[,2]==0,2] <- -1
model_1 <- lmFit(exprs_pair, design_1)


# <---- Model 3: exprs corrected for methylation ~ tissue
# specify design matrix
resid_exprs <- array(0, dim = dim(exprs_pair))
for (index in 1:nrow(exprs_pair)){
  resid_exprs[index,] <- lm(t(exprs_pair[index, ]) ~ t(methyl_pair[index, ]))$resid
}
rownames(resid_exprs) <- rownames(exprs_pair)
model_3 <- lmFit(resid_exprs, design_1)  

```

## Effect size differences  

```{r}
# get effect sizes
beta1 <- coef(model_1[,2])
beta3 <- coef(model_3[,2])
se_beta1 <- se_beta2 <- se_beta3 <- cov_beta13 <- vector("numeric", ngenes)
# <---- get variances of beta1^S tissue effect
se_beta1 <- model_1$sigma*sqrt((solve(t(design_1)%*%design_1))[2,2])
head(cbind(se_beta1,model_1$sigma*sqrt(model_1$cov.coefficients[2,2])))
# <---- get variances of beta2 methylation effect
for (g in 1:length(se_beta2)) {
  design_2g <- model.matrix(~ as.numeric(methyl_pair[g,]))  
  sigma_g <- model_1$sigma[g]
  se_beta2[index] <- sigma_g*sqrt((solve(t(design_2g)%*%design_2g))[2,2])
}
# <---- get variances of beta3 tissue effect
A <- solve(t(design_1)%*%design_1)%*%t(design_1)
contr.vector <- array(c(0,1), dim=c(2,1))
# compute beta3 by hand
for (g in 1:length(se_beta3)) {
  M_g <- t(methyl_pair[g,])
  design_2g <- model.matrix(~ as.numeric(M_g))  
  A_2g <- solve(t(design_2g)%*%design_2g)%*%t(design_2g)
  sigma_g <- model_1$sigma[g]
  var_beta2g <- (se_beta2^2)[g]
  var_part1 <- (se_beta1[g])^2
  var_part2 <- ( A%*%M_g%*%var_beta2g%*%t(M_g)%*%t(A) )[2,2]
  var_part3 <- ( 2*(sigma_g^2)*A%*%t(A_2g)%*%contr.vector%*%t(M_g)%*%t(A) )[2,2]
  se_beta3[g] <- sqrt(var_part1 + var_part2 + var_part3)
  }
# cov(beta1,beta3)
for (g in 1:length(cov_beta13)) {
  M_g <- t(methyl_pair[g,])
  design_2g <- model.matrix(~ as.numeric(M_g))  
  A_2g <- solve(t(design_2g)%*%design_2g)%*%t(design_2g)
  sigma_g <- model_1$sigma[g]
  var_part1 <- (se_beta1[g])^2
  cov_beta13[g] <- var_part1 - (sigma_g^2)*((A %*% t(A_2g) %*%contr.vector%*%t(M_g)%*%t(A))[2,2])
}
  
# cov(beta1-beta3)
cov_diff_sqrt <- sqrt(se_beta1^2 + se_beta3^2 - 2*cov_beta13)
beta_diff <- beta1-beta3

summary(se_beta1^2)
summary(se_beta3^2)
summary(cov_beta13)
summary(cov_diff_sqrt^2)


#col_cov_diff_sqrt <- cov_diff_sqrt/max(cov_diff_sqrt)
plot(x=beta1-beta3, y=cov_diff_sqrt,
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Difference of effect size (human heart vs human kidney)",
     pch=16, cex=.6)

# What is the summary of beta1 and beta3?

plot(beta1, beta3, xlab = "Effect size from lm 1", ylab = "Effect size from lm 2", main = "Effect size comparison of difference in human heart and human kidney")

summary(beta1-beta3)

# How genes have beta1 > beta3?

summary(beta1 > beta3)

# How genes have beta3 > beta1?

summary(beta1 < beta3)


```


Appears that the differences between the effect sizes center at approximately 0 log2 fold change. Also note that standard errors are large for when the effect sizes from before correcting are way smaller than after correcting.


## Run ASH

```{r}
library(ashr)
# mode = a number that corresponds to when there's no difference between 
# the effect sizes. I try for 0, because that seems to be the average here.
# try to experiment different modes and see what happens
fit_ash <- ash(as.vector(beta_diff), cov_diff_sqrt, mode=0)
names(fit_ash$result)
# lfsr is the minimum of PositiveProb and NegativeProb
# look at genes at various cut-off, say svalue < .01 etc. do they make sense?
# ind_sig <- which(fit_ash$result$svalue < .01)
head(fit_ash$result)
#ind_sig <- which(fit_ash$result$svalue < .01)
ind_sig <- (fit_ash$result$svalue < .01)
ind_sig[ind_sig == TRUE] <- "red"
ind_sig[ind_sig == FALSE] <- "black"
ind_sig <- as.data.frame(ind_sig)

make_plot <- cbind(beta1-beta3, cov_diff_sqrt, ind_sig, fit_ash$result$lfdr)

plot(x=make_plot[,1], y=make_plot[,2],
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Testing null hypothesis that the difference in effect sizes = 0",
     pch=16, cex=.6, col = make_plot[,3])
legend("topleft", legend=c("s < 0.01", "s >= 0.01"), col=c("red", "black"), pch=16:16, cex=0.8)

nrow(as.data.frame(ind_sig))






fit_ash <- ash(as.vector(beta_diff), cov_diff_sqrt, mode=1)
names(fit_ash$result)
# lfsr is the minimum of PositiveProb and NegativeProb
# look at genes at various cut-off, say svalue < .01 etc. do they make sense?
# ind_sig <- which(fit_ash$result$svalue < .01)
head(fit_ash$result)
#ind_sig <- which(fit_ash$result$svalue < .01)
ind_sig <- (fit_ash$result$svalue < .01)
ind_sig[ind_sig == TRUE] <- "red"
ind_sig[ind_sig == FALSE] <- "black"
ind_sig <- as.data.frame(ind_sig)


make_plot <- cbind(beta1-beta3, cov_diff_sqrt, ind_sig)

plot(x=make_plot[,1], y=make_plot[,2],
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Difference of effect size (human heart vs human kidney)",
     pch=16, cex=.6, col = make_plot[,3])


fit_ash <- ash(as.vector(beta_diff), cov_diff_sqrt, mode=-1)
names(fit_ash$result)
# lfsr is the minimum of PositiveProb and NegativeProb
# look at genes at various cut-off, say svalue < .01 etc. do they make sense?
# ind_sig <- which(fit_ash$result$svalue < .01)
head(fit_ash$result)
#ind_sig <- which(fit_ash$result$svalue < .01)
ind_sig <- (fit_ash$result$svalue < .01)
ind_sig[ind_sig == TRUE] <- "red"
ind_sig[ind_sig == FALSE] <- "black"
ind_sig <- as.data.frame(ind_sig)


make_plot <- cbind(beta1-beta3, cov_diff_sqrt, ind_sig)

plot(x=make_plot[,1], y=make_plot[,2],
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Difference of effect size (human heart vs human kidney)",
     pch=16, cex=.6, col = make_plot[,3])




fit_ash <- ash(as.vector(beta_diff), cov_diff_sqrt, mode=2)
names(fit_ash$result)
# lfsr is the minimum of PositiveProb and NegativeProb
# look at genes at various cut-off, say svalue < .01 etc. do they make sense?
# ind_sig <- which(fit_ash$result$svalue < .01)
head(fit_ash$result)
#ind_sig <- which(fit_ash$result$svalue < .01)
ind_sig <- (fit_ash$result$svalue < .01)
ind_sig[ind_sig == TRUE] <- "red"
ind_sig[ind_sig == FALSE] <- "black"
ind_sig <- as.data.frame(ind_sig)


make_plot <- cbind(beta1-beta3, cov_diff_sqrt, ind_sig)

plot(x=make_plot[,1], y=make_plot[,2],
     ylab = "Standard error of the difference",
     xlab = "Difference in effect size from 2 lms", 
     main = "Difference of effect size (human heart vs human kidney)",
     pch=16, cex=.6, col = make_plot[,3])



```


A commonly used diagnostics in checking ash results is to compare the raw effect sizes versus the posterior mean estimated by ash. This gives you an idea how much shrinkage that was done.
Results here: Observe shrinkage of effect size for positive effect size but not for negative.

```{r}
plot(x=fit_ash$result$betahat,
     y=fit_ash$result$PosteriorMean,
     xlab = "Raw difference", ylab = "Posterior estimate", 
     main = "Difference in effect size before and after ASH",
     pch =16,cex=.6)
abline(0,1,col="red")
```






