---
title: "Non_orthologous_cpgs_for_orth_genes"
author: "Lauren Blake"
date: "October 30, 2017"
output: html_document
---

In this script, we will perform a joint analysis of methylation and gene expression levels in humans and chimpanzees only. We chose only these two species because the number of genes that we had methylation values for around the promoter was quite low when we required orthology between humans, chimps, and rhesus. Permutations are computationally intensive, so we have an RMarkdown example with permutations, an R script to run the code and permutations, and this script, in which we will focus on the beta coefficients. We will make the plots of the Beta coefficients before and after regressing out methylation for each set of tissues. 

We will make DE decisions based on FSR rather than FDR. 


```{r}
# Load libraries/data

library("edgeR")
library("limma")
library("plyr")
library("ggplot2")

cpm.voom.cyclic <- readRDS("../data/human_chimp_orth_cpm_voom_cyclic.rds")
exp_methyl <- read.table("../data/human_chimp_orth_exp_methyl_7725_hum.txt", header = T, stringsAsFactors = F)
samples <- read.table("../data/human_chimp_orth_new_sample_info.txt", header = T,  stringsAsFactors = F)

# Get sample info
species <- as.data.frame(samples[,4])
tissue <- as.data.frame(samples[,5])
RIN <- as.data.frame(samples[,6])
```

# Check the variances  

## Human Heart v Kidney

```{r}
# Human Heart v Kidney
# First, pick out the genes that are DE (you run the entire analysis only on genes that are DE)
chimp_human_heart <- c(17, 20, 21, 24, 25, 28, 29)

#chimp_human_heart <- c(1, 2, 5, 6, 9, 10, 13, 14)

FSR_level <- 0.05
FDR_level <- FSR_level
tissue <- tissue[chimp_human_heart,]
tissue_no_extra <- droplevels.factor(tissue, "liver")


design <- model.matrix(~ as.factor(tissue_no_extra) + RIN[chimp_human_heart,])
fit_all <- lmFit(cpm.voom.cyclic[,chimp_human_heart], design)
fit_all <- eBayes(fit_all)


# Run ASH

library(ashr)

# Prepare the data for ASH
tests <- colnames(fit_all$coefficients)
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Perform multiple testing correction with adaptive shrinkage (ASH) 
 #
 # x - object MArrayLM from eBayes output
 # coef - coefficient tested by eBayes

run_ash <- function(x, coef){
  #stopifnot(class(x) == "MArrayLM", coef %in% colnames(x$coefficients),
  #             length(unique(x$df.total) == 1))
  result <- ash(betahat = x$coefficients[, coef], sebetahat = x$stdev.unscaled[, coef] * sqrt(x$s2.post), df = x$df.total[1])
  return(result)
}

get_results <- function(x, number = nrow(x$coefficients), sort.by = "none",
                        ...) {
  # x - object MArrayLM from eBayes output
  # ... - additional arguments passed to topTable
  stopifnot(class(x) == "MArrayLM")
  results <- topTable(x, number = number, sort.by = sort.by, ...)
  return(results)
}

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
for (test in tests) {
  # Extract limma results
  results[[test]] <- get_results(fit_all, coef = test)
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit_all, coef = test)
  results[[test]] <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)
}

new_results <- results[["as.factor(tissue_no_extra)kidney"]]
summary(new_results)
plot(new_results$adj.P.Val, new_results$qvalue)


# Find the DE genes
HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust="BH", number=Inf, sort.by="none")
HvC_Heart_fit_all_5perc <- HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val < FSR_level), ]
HvC_Heart_fit_all_5perc <- new_results[which(new_results$qvalue < FSR_level), ]
HvC_Heart_fit_all_5perc <- new_results[which(new_results$svalue < FSR_level), ]


human_chimp_heart <-  rownames(exp_methyl) %in% HvC_Heart_fit_all_5perc$genes
human_chimp_heart <- as.data.frame(human_chimp_heart)
counts_genes_in <- cbind(exp_methyl, human_chimp_heart)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart == "TRUE")
counts_genes_in_cutoff <- counts_genes_in_cutoff[,1:79]  
expression_values_only <- counts_genes_in_cutoff[,2:32]
methylation_values_only <- counts_genes_in_cutoff[,49:79]

## Obtain corrected data (regression for scenario 2) by regressing out RIN on a gene-by-gene basis
resid_methyl <- array(0, dim = c(nrow(expression_values_only), length(chimp_human_heart))) 
expression_values <- as.data.frame(expression_values_only)
methylation_values <- as.data.frame(methylation_values_only)
for (i in 1:nrow(expression_values_only)){
  resid_methyl[i,] <- lm(t(expression_values_only[i,chimp_human_heart]) ~ t(methylation_values_only[i,chimp_human_heart]))$resid
}
rownames(resid_methyl) <- rownames(expression_values_only)

# Scenario 1

fit1 <- lmFit(expression_values_only[, chimp_human_heart], design)
fit1 <- eBayes(fit1)
HvC_Heart_fit1 = topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")
summary(HvC_Heart_fit1$adj.P.Val < FDR_level)

# Scenario 2

fit2 <- lmFit(resid_methyl, design)
fit2 <- eBayes(fit2)
HvC_Heart_fit2 = topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")
summary(HvC_Heart_fit2$adj.P.Val < FDR_level)

# Bind DE from fit1 and DE from fit2 together

HvC_Heart_fits12 <- as.data.frame(cbind(rownames(HvC_Heart_fit1), fit1$coefficients[,2], HvC_Heart_fit1$adj.P.Val, fit2$coefficients[,2], HvC_Heart_fit2$adj.P.Val), stringsAsFactors = FALSE)
HvC_Heart_fits12[,2] <- as.numeric(HvC_Heart_fits12[,2])
HvC_Heart_fits12[,3] <- as.numeric(HvC_Heart_fits12[,3])
HvC_Heart_fits12[,4] <- as.numeric(HvC_Heart_fits12[,4])
HvC_Heart_fits12[,5] <- as.numeric(HvC_Heart_fits12[,5])
colnames(HvC_Heart_fits12) <- c("genes", "fit1_species_coef", "fit1_FDR", "fit2_species_coef", "fit2_FDR")

# Find the number of DE genes
DE_both <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_both)

DE_before <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR > FDR_level),]
nrow(DE_before)

DE_after <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR > FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_after)

# Make the plot of betas

ggplot(HvC_Heart_fits12, aes(fit1_species_coef, fit2_species_coef)) + geom_point()
```


```{r}
library(ashr)
# Chimp Kidney v Liver
# First, pick out the genes that are DE (you run the entire analysis only on genes that are DE)
tissue <- as.data.frame(samples[,5])
chimp_human_heart <- c(17, 18, 21, 22, 25, 26, 29, 30)

# chimp_human_heart <- c(2, 3, 6, 7, 10, 11, 14, 15)

FSR_level <- 0.05
tissue <- tissue[chimp_human_heart,]
tissue_no_extra <- droplevels.factor(tissue, "lung")


design <- model.matrix(~ as.factor(tissue_no_extra) + RIN[chimp_human_heart,])
fit_all <- lmFit(cpm.voom.cyclic[,chimp_human_heart], design)
fit_all <- eBayes(fit_all)


# Run ASH

# Prepare the data for ASH
tests <- colnames(fit_all$coefficients)
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Perform multiple testing correction with adaptive shrinkage (ASH) 
 #
 # x - object MArrayLM from eBayes output
 # coef - coefficient tested by eBayes

run_ash <- function(x, coef){
  #stopifnot(class(x) == "MArrayLM", coef %in% colnames(x$coefficients),
  #             length(unique(x$df.total) == 1))
  result <- ash(betahat = x$coefficients[, coef], sebetahat = x$stdev.unscaled[, coef] * sqrt(x$s2.post), df = x$df.total[1])
  return(result)
}

get_results <- function(x, number = nrow(x$coefficients), sort.by = "none",
                        ...) {
  # x - object MArrayLM from eBayes output
  # ... - additional arguments passed to topTable
  stopifnot(class(x) == "MArrayLM")
  results <- topTable(x, number = number, sort.by = sort.by, ...)
  return(results)
}

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
  # Extract limma results
  results[[test]] <- get_results(fit_all, coef = tests[2])
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit_all, coef = tests[2])
  results <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

# Find the genes < FSR
#HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust="BH", number=Inf, sort.by="none")
#HvC_Heart_fit_all_5perc <- HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val < FSR_level), ]
#HvC_Heart_fit_all_5perc <- results[which(results$qvalue < FSR_level), ]
HvC_Heart_fit_all_5perc <- results[which(new_results$svalue < FSR_level), ]


human_chimp_heart <-  rownames(exp_methyl) %in% HvC_Heart_fit_all_5perc$genes
human_chimp_heart <- as.data.frame(human_chimp_heart)
counts_genes_in <- cbind(exp_methyl, human_chimp_heart)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart == "TRUE")
counts_genes_in_cutoff <- counts_genes_in_cutoff[,1:79]  
expression_values_only <- counts_genes_in_cutoff[,2:32]
methylation_values_only <- counts_genes_in_cutoff[,49:79]
dim(expression_values_only)

## Obtain corrected data (regression for scenario 2) by regressing out RIN on a gene-by-gene basis
resid_methyl <- array(0, dim = c(nrow(expression_values_only), length(chimp_human_heart))) 
expression_values <- as.data.frame(expression_values_only)
methylation_values <- as.data.frame(methylation_values_only)
for (i in 1:nrow(expression_values_only)){
  resid_methyl[i,] <- lm(t(expression_values_only[i,chimp_human_heart]) ~ t(methylation_values_only[i,chimp_human_heart]))$resid
}
rownames(resid_methyl) <- rownames(expression_values_only)

# Scenario 1

fit1 <- lmFit(expression_values_only[, chimp_human_heart], design)
fit1 <- eBayes(fit1)

  results <- get_results(fit1, coef = tests[2])
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit1, coef = tests[2])
  results_fit1 <- cbind(results, sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)


# Scenario 2

fit2 <- lmFit(resid_methyl, design)
fit2 <- eBayes(fit2)

results <- get_results(fit1, coef = tests[2])
  # Add mutliple testing correction with ASH
output_ash <- run_ash(fit2, coef = tests[2])
results_fit2 <- cbind(results, sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

# Bind DE from fit1 and DE from fit2 together

HvC_Heart_fits12 <- as.data.frame(cbind(rownames(results_fit1), results_fit1$svalue, results_fit2$svalue), stringsAsFactors = FALSE)
HvC_Heart_fits12[,2] <- as.numeric(HvC_Heart_fits12[,2])
HvC_Heart_fits12[,3] <- as.numeric(HvC_Heart_fits12[,3])
colnames(HvC_Heart_fits12) <- c("genes", "fit1_FDR", "fit2_FDR")

DE_both <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_both)

DE_before <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR > FDR_level),]
nrow(DE_before)

DE_after <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR > FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_after)

```

### Make this into a function (where you give it the FSR and the positions)

```{r}
integration_ash <- function(chimp_human_heart, FSR_level){

library(ashr)
FDR_level <- FSR_level
tissue <- as.data.frame(samples[,5])
tissue <- tissue[chimp_human_heart,]
tissue_no_extra <- droplevels.factor(tissue)


design <- model.matrix(~ as.factor(tissue_no_extra) + RIN[chimp_human_heart,])
fit_all <- lmFit(cpm.voom.cyclic[,chimp_human_heart], design)
fit_all <- eBayes(fit_all)


# Run ASH

# Prepare the data for ASH
tests <- colnames(fit_all$coefficients)
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Perform multiple testing correction with adaptive shrinkage (ASH) 
 #
 # x - object MArrayLM from eBayes output
 # coef - coefficient tested by eBayes

run_ash <- function(x, coef){
  #stopifnot(class(x) == "MArrayLM", coef %in% colnames(x$coefficients),
  #             length(unique(x$df.total) == 1))
  result <- ash(betahat = x$coefficients[, coef], sebetahat = x$stdev.unscaled[, coef] * sqrt(x$s2.post), df = x$df.total[1])
  return(result)
}

get_results <- function(x, number = nrow(x$coefficients), sort.by = "none",
                        ...) {
  # x - object MArrayLM from eBayes output
  # ... - additional arguments passed to topTable
  stopifnot(class(x) == "MArrayLM")
  results <- topTable(x, number = number, sort.by = sort.by, ...)
  return(results)
}

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
  # Extract limma results
  results <- get_results(fit_all, coef = tests[2])
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit_all, coef = tests[2])
  results <- cbind(results, sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

# Find the genes < FSR
#HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust="BH", number=Inf, sort.by="none")
#HvC_Heart_fit_all_5perc <- HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val < FSR_level), ]
#HvC_Heart_fit_all_5perc <- results[which(results$qvalue < FSR_level), ]
HvC_Heart_fit_all_5perc <- results[which(results$svalue < FSR_level), ]


human_chimp_heart <-  rownames(exp_methyl) %in% HvC_Heart_fit_all_5perc$genes
human_chimp_heart <- as.data.frame(human_chimp_heart)
counts_genes_in <- cbind(exp_methyl, human_chimp_heart)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart == "TRUE")
counts_genes_in_cutoff <- counts_genes_in_cutoff[,1:79]  
expression_values_only <- counts_genes_in_cutoff[,2:32]
methylation_values_only <- counts_genes_in_cutoff[,49:79]
dim(expression_values_only)

## Obtain corrected data (regression for scenario 2) by regressing out RIN on a gene-by-gene basis
resid_methyl <- array(0, dim = c(nrow(expression_values_only), length(chimp_human_heart))) 
expression_values <- as.data.frame(expression_values_only)
methylation_values <- as.data.frame(methylation_values_only)
for (i in 1:nrow(expression_values_only)){
  resid_methyl[i,] <- lm(t(expression_values_only[i,chimp_human_heart]) ~ t(methylation_values_only[i,chimp_human_heart]))$resid
}
rownames(resid_methyl) <- rownames(expression_values_only)

# Scenario 1

fit1 <- lmFit(expression_values_only[, chimp_human_heart], design)
fit1 <- eBayes(fit1)

  results <- get_results(fit1, coef = tests[2])
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit1, coef = tests[2])
  results_fit1 <- cbind(results, sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)


# Scenario 2

fit2 <- lmFit(resid_methyl, design)
fit2 <- eBayes(fit2)

results <- get_results(fit2, coef = tests[2])
  # Add mutliple testing correction with ASH
output_ash <- run_ash(fit2, coef = tests[2])
results_fit2 <- cbind(results, sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

# Bind DE from fit1 and DE from fit2 together

HvC_Heart_fits12 <- as.data.frame(cbind(rownames(results_fit1), results_fit1$svalue, results_fit2$svalue), stringsAsFactors = FALSE)
HvC_Heart_fits12[,2] <- as.numeric(HvC_Heart_fits12[,2])
HvC_Heart_fits12[,3] <- as.numeric(HvC_Heart_fits12[,3])
colnames(HvC_Heart_fits12) <- c("genes", "fit1_FDR", "fit2_FDR")

DE_both <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_both)

DE_before <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR > FDR_level),]
nrow(DE_before)

DE_after <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR > FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_after)

important_returns <- rbind(nrow(DE_both), nrow(DE_before), nrow(DE_after), nrow(expression_values_only))
return(important_returns)
}

# Human vs chimp heart
#integration_ash(c(1, 5, 9, 13, 20, 24, 28), 0.05)
#integration_ash(c(1, 5, 9, 13, 20, 24, 28), 0.1)

# Human vs chimp kidney
#integration_ash(c(2, 6, 10, 14, 17, 21, 25, 29), 0.05)
#integration_ash(c(2, 6, 10, 14, 17, 21, 25, 29), 0.1)

# Human vs chimp liver
#integration_ash(c(3, 7, 11, 15, 18, 22, 26, 30), 0.05)
#integration_ash(c(3, 7, 11, 15, 18, 22, 26, 30), 0.1)

# Human vs chimp lung
#integration_ash(c(4, 8, 12, 16, 19, 23, 27, 31), 0.05)
#integration_ash(c(4, 8, 12, 16, 19, 23, 27, 31), 0.1)

# Human heart-kidney
integration_ash(c(17, 20, 21, 24, 25, 28, 29), 0.05)
integration_ash(c(17, 20, 21, 24, 25, 28, 29), 0.10)
# Chimp heart-kidney
integration_ash(c(1, 2, 5, 6, 9, 10, 13, 14), 0.05)
integration_ash(c(1, 2, 5, 6, 9, 10, 13, 14), 0.1)

# Human heart-liver

# Chimp heart-liver
integration_ash(c(1, 3, 5, 7, 9, 11, 13, 15), 0.05)
integration_ash(c(1, 3, 5, 7, 9, 11, 13, 15), 0.1)

# Human heart-lung
integration_ash(c(19, 20, 23, 24, 27, 28, 31), 0.05)
integration_ash(c(19, 20, 23, 24, 27, 28, 31), 0.1)
# Chimp heart-lung
integration_ash(c(1, 4, 5, 8, 9, 12, 13, 16), 0.05)
integration_ash(c(1, 4, 5, 8, 9, 12, 13, 16), 0.1)

# Human kidney-liver
integration_ash(c(17, 18, 21, 22, 25, 26, 29, 30), 0.05)
integration_ash(c(17, 18, 21, 22, 25, 26, 29, 30), 0.1)
# Chimp kidney-liver
integration_ash(c(2, 3, 6, 7, 10, 11, 14, 15), 0.05)
integration_ash(c(2, 3, 6, 7, 10, 11, 14, 15), 0.1)

# Human kidney-lung
integration_ash(c(17, 19, 21, 23, 25, 27, 29, 31), 0.05)
integration_ash(c(17, 19, 21, 23, 25, 27, 29, 31), 0.1)
# Chimp kidney-lung
integration_ash(c(2, 4, 6, 8, 10, 12, 14, 16), 0.05)
integration_ash(c(2, 4, 6, 8, 10, 12, 14, 16), 0.1)

# Human liver-lung
integration_ash(c(18, 19, 22, 23, 26, 27, 30, 31), 0.05)
integration_ash(c(18, 19, 22, 23, 26, 27, 30, 31), 0.1)
# Chimp liver-lung
integration_ash(c(3, 4, 7, 8, 11, 12, 15, 16), 0.05)
integration_ash(c(3, 4, 7, 8, 11, 12, 15, 16), 0.1)
```

library("edgeR")
library("limma")
library("plyr")

cpm.voom.cyclic <- readRDS("./Methylation_integration/human_chimp_orth_cpm_voom_cyclic.rds")
exp_methyl <- read.table("./Methylation_integration/human_chimp_orth_exp_methyl_7725_hum.txt", header = T, stringsAsFactors = F)
samples <- read.table("./Methylation_integration/human_chimp_orth_new_sample_info.txt", header = T,  stringsAsFactors = F)

species <- as.data.frame(samples[,4])
tissue <- as.data.frame(samples[,5])
RIN <- as.data.frame(samples[,6])
library(ashr)


integration_ash <- function(chimp_human_heart, FSR_level){

tissue <- as.data.frame(samples[,5])
tissue <- tissue[chimp_human_heart,]
tissue_no_extra <- droplevels.factor(tissue)


design <- model.matrix(~ as.factor(tissue_no_extra) + RIN[chimp_human_heart,])
fit_all <- lmFit(cpm.voom.cyclic[,chimp_human_heart], design)
fit_all <- eBayes(fit_all)


# Run ASH

# Prepare the data for ASH
tests <- colnames(fit_all$coefficients)
results <- vector(length = length(tests), mode = "list")
names(results) <- tests

# Perform multiple testing correction with adaptive shrinkage (ASH) 
 #
 # x - object MArrayLM from eBayes output
 # coef - coefficient tested by eBayes

run_ash <- function(x, coef){
  #stopifnot(class(x) == "MArrayLM", coef %in% colnames(x$coefficients),
  #             length(unique(x$df.total) == 1))
  result <- ash(betahat = x$coefficients[, coef], sebetahat = x$stdev.unscaled[, coef] * sqrt(x$s2.post), df = x$df.total[1])
  return(result)
}

get_results <- function(x, number = nrow(x$coefficients), sort.by = "none",
                        ...) {
  # x - object MArrayLM from eBayes output
  # ... - additional arguments passed to topTable
  stopifnot(class(x) == "MArrayLM")
  results <- topTable(x, number = number, sort.by = sort.by, ...)
  return(results)
}

# Get lfsr, lfdr, s value, q value, and a beta_est value. 
  # Extract limma results
  results[[test]] <- get_results(fit_all, coef = tests[2])
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit_all, coef = tests[2])
  results <- cbind(results[[test]], sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

# Find the genes < FSR
#HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust="BH", number=Inf, sort.by="none")
#HvC_Heart_fit_all_5perc <- HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val < FSR_level), ]
#HvC_Heart_fit_all_5perc <- results[which(results$qvalue < FSR_level), ]
HvC_Heart_fit_all_5perc <- results[which(results$svalue < FSR_level), ]


human_chimp_heart <-  rownames(exp_methyl) %in% HvC_Heart_fit_all_5perc$genes
human_chimp_heart <- as.data.frame(human_chimp_heart)
counts_genes_in <- cbind(exp_methyl, human_chimp_heart)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart == "TRUE")
counts_genes_in_cutoff <- counts_genes_in_cutoff[,1:79]  
expression_values_only <- counts_genes_in_cutoff[,2:32]
methylation_values_only <- counts_genes_in_cutoff[,49:79]
dim(expression_values_only)

## Obtain corrected data (regression for scenario 2) by regressing out RIN on a gene-by-gene basis
resid_methyl <- array(0, dim = c(nrow(expression_values_only), length(chimp_human_heart))) 
expression_values <- as.data.frame(expression_values_only)
methylation_values <- as.data.frame(methylation_values_only)
for (i in 1:nrow(expression_values_only)){
  resid_methyl[i,] <- lm(t(expression_values_only[i,chimp_human_heart]) ~ t(methylation_values_only[i,chimp_human_heart]))$resid
}
rownames(resid_methyl) <- rownames(expression_values_only)

# Scenario 1

fit1 <- lmFit(expression_values_only[, chimp_human_heart], design)
fit1 <- eBayes(fit1)

  results <- get_results(fit1, coef = tests[2])
  # Add mutliple testing correction with ASH
  output_ash <- run_ash(fit1, coef = tests[2])
  results_fit1 <- cbind(results, sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)


# Scenario 2

fit2 <- lmFit(resid_methyl, design)
fit2 <- eBayes(fit2)

results <- get_results(fit1, coef = tests[2])
  # Add mutliple testing correction with ASH
output_ash <- run_ash(fit2, coef = tests[2])
results_fit2 <- cbind(results, sebetahat = output_ash$data$s, lfsr = output_ash$result$lfsr,
                           lfdr = output_ash$result$lfdr, qvalue = output_ash$result$qvalue,
                           svalue = output_ash$result$svalue, beta_est = output_ash$result$PosteriorMean, se_est =
                             output_ash$result$PosteriorSD)

# Bind DE from fit1 and DE from fit2 together

HvC_Heart_fits12 <- as.data.frame(cbind(rownames(results_fit1), results_fit1$svalue, results_fit2$svalue), stringsAsFactors = FALSE)
HvC_Heart_fits12[,2] <- as.numeric(HvC_Heart_fits12[,2])
HvC_Heart_fits12[,3] <- as.numeric(HvC_Heart_fits12[,3])
colnames(HvC_Heart_fits12) <- c("genes", "fit1_FDR", "fit2_FDR")

DE_both <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_both)

DE_before <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR > FDR_level),]
nrow(DE_before)

DE_after <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR > FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_after)

important_returns <- rbind(nrow(DE_both), nrow(DE_before), nrow(DE_after), nrow(expression_values_only))
return(important_returns)
}



results_perm <- integration_ash(c(17, 20, 21, 24, 25, 28, 29), 0.05)

