<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />

<meta name="date" content="2016-08-22" />

<title>DE_genes_pairwise_ts_specific</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="libs/highlight/textmate.css"
      type="text/css" />
<script src="libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RegulatoryEvolutionInPrimates</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Reg_Evo_Primates">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">DE_genes_pairwise_ts_specific</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>August 22, 2016</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#part-1-find-genes-that-are-de-between-different-tissues-or-species-pairwise.">Part 1: Find genes that are DE between different tissues or species, pairwise.</a></li>
<li><a href="#part-2-tissue-specific-expression-in-and-across-species">Part 2: Tissue specific expression in and across species</a></li>
</ul>
</div>

<p>The goal of this script is to find a list of differentially expressed genes between tissues and between species. The majority of the code for this section was written by Julien Roux.</p>
<p>Note: the expression data used here is the final filtering scheme of only keeping genes with log2(CPM) &gt; 1.5 in at least 24 samples.</p>
<div id="part-1-find-genes-that-are-de-between-different-tissues-or-species-pairwise." class="section level3">
<h3>Part 1: Find genes that are DE between different tissues or species, pairwise.</h3>
<pre class="r"><code># Load libraries

library(&quot;gplots&quot;)</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)
library(&quot;RColorBrewer&quot;)
library(&quot;scales&quot;)
library(&quot;edgeR&quot;)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;R.utils&quot;)</code></pre>
<pre><code>Loading required package: R.oo</code></pre>
<pre><code>Loading required package: R.methodsS3</code></pre>
<pre><code>R.methodsS3 v1.7.1 (2016-02-15) successfully loaded. See ?R.methodsS3 for help.</code></pre>
<pre><code>R.oo v1.20.0 (2016-02-17) successfully loaded. See ?R.oo for help.</code></pre>
<pre><code>
Attaching package: &#39;R.oo&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:methods&#39;:

    getClasses, getMethods</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    attach, detach, gc, load, save</code></pre>
<pre><code>R.utils v2.3.0 (2016-04-13) successfully loaded. See ?R.utils for help.</code></pre>
<pre><code>
Attaching package: &#39;R.utils&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    timestamp</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    cat, commandArgs, getOption, inherits, isOpen, parse, warnings</code></pre>
<pre class="r"><code>library(&quot;plyr&quot;)
library(&quot;limma&quot;)
library(&quot;VennDiagram&quot;)</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre><code>Loading required package: futile.logger</code></pre>
<pre class="r"><code>source(&quot;~/Reg_Evo_Primates/ashlar-trial/analysis/functions.R&quot;)

# Load colors 

colors &lt;- colorRampPalette(c(brewer.pal(9, &quot;Blues&quot;)[1],brewer.pal(9, &quot;Blues&quot;)[9]))(100)
pal &lt;- c(brewer.pal(9, &quot;Set1&quot;), brewer.pal(8, &quot;Set2&quot;), brewer.pal(12, &quot;Set3&quot;))

# Import samples

samples &lt;- read.delim(&quot;~/Reg_Evo_Primates/ashlar-trial/data/Sample_info_RNAseq_limma.txt&quot;)

# Eliminate H1H
samples &lt;- samples[-17,]
dim(samples)</code></pre>
<pre><code>[1] 47  4</code></pre>
<pre class="r"><code># Make labels

labels &lt;- paste(samples$Species, samples$Tissue, sep=&quot; &quot;)</code></pre>
<pre class="r"><code># Load the data and normalize it (TMM+CPM+random effect+voom+cyclic loess normalization)

counts_genes_in_cutoff &lt;- read.delim(&quot;~/Reg_Evo_Primates/ashlar-trial/data/counts_12184.txt&quot;)

dim(counts_genes_in_cutoff)</code></pre>
<pre><code>[1] 12184    47</code></pre>
<pre class="r"><code># Take the TMM of the genes that meet the criteria

dge_in_cutoff &lt;- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff &lt;- calcNormFactors(dge_in_cutoff)

# Perform voom

## Voom requires a design matrix as input                                                                                                                               
## To make contrasts easier to formulate, we rename factors species and tissue in a single factor 

condition &lt;- factor(paste(samples$Species,samples$Tissue,sep=&quot;.&quot;))
design &lt;- model.matrix(~ 0 + condition)
colnames(design) &lt;- gsub(&quot;condition&quot;, &quot;&quot;, dput(colnames(design)))</code></pre>
<pre><code>c(&quot;conditionChimp.heart&quot;, &quot;conditionChimp.kidney&quot;, &quot;conditionChimp.liver&quot;, 
&quot;conditionChimp.lung&quot;, &quot;conditionHuman.heart&quot;, &quot;conditionHuman.kidney&quot;, 
&quot;conditionHuman.liver&quot;, &quot;conditionHuman.lung&quot;, &quot;conditionRhesus.heart&quot;, 
&quot;conditionRhesus.kidney&quot;, &quot;conditionRhesus.liver&quot;, &quot;conditionRhesus.lung&quot;
)</code></pre>
<pre class="r"><code># Voom with individual as a random variable

cpm.voom.cyclic &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=T)</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/normalization-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>corfit &lt;- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)

# corfit$consensus = 0.2006188

# Final voom on filtered data

cpm.voom.cyclic.ran &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;cyclicloess&quot;, plot=TRUE, block=samples$Individual, correlation=corfit$consensus)</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/normalization-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>We saw in the technical factor analysis that the hours post-mortem that the sample was collected was confounded by species. This could affect the quality of RNA. We will use RIN score as a proxy for RNA quality. Let see how adding RIN score into the model (via the design matrix) affects the amount of tissue-specific DE genes.</p>
<div id="initial-running-of-limma" class="section level4">
<h4>Initial running of limma</h4>
<pre class="r"><code># Retrieve RIN score for each sample
RNA_seq_info &lt;- read.csv(&quot;~/Reg_Evo_Primates/ashlar-trial/data/RNA_seq_info.csv&quot;)
RIN &lt;- as.data.frame(RNA_seq_info[,22])
RIN &lt;- RIN[-31,]

# Remake design matrix

condition &lt;- factor(paste(samples$Species,samples$Tissue,sep=&quot;.&quot;))
condition &lt;- as.data.frame(condition)
condition &lt;- condition[-31,]
design &lt;- model.matrix(~ 0 + condition + RIN)
colnames(design) &lt;- gsub(&quot;condition&quot;, &quot;&quot;, dput(colnames(design)))</code></pre>
<pre><code>c(&quot;conditionChimp.heart&quot;, &quot;conditionChimp.kidney&quot;, &quot;conditionChimp.liver&quot;, 
&quot;conditionChimp.lung&quot;, &quot;conditionHuman.heart&quot;, &quot;conditionHuman.kidney&quot;, 
&quot;conditionHuman.liver&quot;, &quot;conditionHuman.lung&quot;, &quot;conditionRhesus.heart&quot;, 
&quot;conditionRhesus.kidney&quot;, &quot;conditionRhesus.liver&quot;, &quot;conditionRhesus.lung&quot;, 
&quot;RIN&quot;)</code></pre>
<pre class="r"><code># Run lmFit and eBayes in limma (we have to take out sample 31 because its RIN score was not recorded)
samples &lt;- samples[-31,]

fit.cpm.voom.cyclic.ran &lt;- lmFit(cpm.voom.cyclic.ran[,-31], design, block=samples$Individual, correlation=corfit$consensus)
fit.cpm.voom.cyclic.ran &lt;- eBayes(fit.cpm.voom.cyclic.ran)

# MA Plots
## If &#39;MA&#39; is an &#39;MArrayLM&#39; object, then the plot is a fitted model MA-plot in which the estimated coefficient is on the y-axis and the average A-value is on the x-axis.                
limma::plotMA(fit.cpm.voom.cyclic.ran, array=1, xlab=&quot;average coefficient&quot;, ylab=&quot;estimated coefficient&quot;)</code></pre>
<pre><code>Warning in plot.window(...): &quot;array&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in plot.xy(xy, type, ...): &quot;array&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in axis(side = side, at = at, labels = labels, ...): &quot;array&quot; is not
a graphical parameter

Warning in axis(side = side, at = at, labels = labels, ...): &quot;array&quot; is not
a graphical parameter</code></pre>
<pre><code>Warning in box(...): &quot;array&quot; is not a graphical parameter</code></pre>
<pre><code>Warning in title(...): &quot;array&quot; is not a graphical parameter</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/basic%20limma%20with%20RIN-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## - Potential caveat: variances could be different between human, chimp and rhesus (see Gordon Smyth email, 7 June 2013).                                                               
##  We look at the standard error for each condition                                                    
hist(fit.cpm.voom.cyclic.ran$stdev.unscaled * fit.cpm.voom.cyclic.ran$sigma, breaks=100)</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/basic%20limma%20with%20RIN-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(log2(fit.cpm.voom.cyclic.ran$stdev.unscaled * fit.cpm.voom.cyclic.ran$sigma), breaks=100)</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/basic%20limma%20with%20RIN-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(log2(fit.cpm.voom.cyclic.ran$stdev.unscaled * fit.cpm.voom.cyclic.ran$sigma))</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/basic%20limma%20with%20RIN-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## This seems to be pretty comparable between conditions. The human heart is higher, probably because of H1H missing and H3H with a bit strange behavior                                     
stderror &lt;- log2(fit.cpm.voom.cyclic.ran$stdev.unscaled * fit.cpm.voom.cyclic.ran$sigma)
boxplot(list(stderror[,1:4], stderror[,5:8], stderror[,9:12]))</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/basic%20limma%20with%20RIN-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## A bit higher for human, and a bit lower for rhesus                                                                                                                                    
boxplot(list(stderror[,2:4], stderror[,6:8], stderror[,8:12])) ## excluding heart samples  </code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/basic%20limma%20with%20RIN-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-between-2-tissues-in-the-same-species" class="section level4">
<h4>Differential expression between 2 tissues (in the same species)</h4>
<pre class="r"><code># Make the contrast matrix

cm4 &lt;- makeContrasts(
                     ChimpKidneyvsLiver = Chimp.kidney-Chimp.liver,
                     HumanKidneyvsLiver = Human.kidney-Human.liver,
                     RhesusKidneyvsLiver = Rhesus.kidney-Rhesus.liver,
                     ChimpKidneyvsLung = Chimp.kidney-Chimp.lung,
                     HumanKidneyvsLung = Human.kidney-Human.lung,
                     RhesusKidneyvsLung = Rhesus.kidney-Rhesus.lung,
                     ChimpKidneyvsHeart = Chimp.kidney-Chimp.heart,
                     HumanKidneyvsHeart = Human.kidney-Human.heart,
                     RhesusKidneyvsHeart = Rhesus.kidney-Rhesus.heart,
                     ChimpHeartvsLiver = Chimp.heart-Chimp.liver,
                     HumanHeartvsLiver = Human.heart-Human.liver,
                     RhesusHeartvsLiver = Rhesus.heart-Rhesus.liver,
                     ChimpHeartvsLung = Chimp.heart-Chimp.lung,
                     HumanHeartvsLung = Human.heart-Human.lung,
                     RhesusHeartvsLung = Rhesus.heart-Rhesus.lung,
                     ChimpLungvsLiver = Chimp.lung-Chimp.liver,
                     HumanLungvsLiver = Human.lung-Human.liver,
                     RhesusLungvsLiver = Rhesus.lung-Rhesus.liver,
                     levels=design
                     )

# Run the linear model

fit4 &lt;- contrasts.fit(fit.cpm.voom.cyclic.ran, cm4)
fit4 &lt;- eBayes(fit4)

# Order by BH value
top4 &lt;- list(ChimpKidneyvsLiver=topTable(fit4, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpKidneyvsLung=topTable(fit4, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpKidneyvsHeart=topTable(fit4, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpHeartvsLiver=topTable(fit4, coef=10, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpHeartvsLung=topTable(fit4, coef=13, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpLungvsLiver=topTable(fit4, coef=16, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanKidneyvsLiver=topTable(fit4, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanKidneyvsLung=topTable(fit4, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanKidneyvsHeart=topTable(fit4, coef=8, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanHeartvsLiver=topTable(fit4, coef=11, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanHeartvsLung=topTable(fit4, coef=14, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanLungvsLiver=topTable(fit4, coef=17, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusKidneyvsLiver=topTable(fit4, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusKidneyvsLung=topTable(fit4, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusKidneyvsHeart=topTable(fit4, coef=9, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusHeartvsLiver=topTable(fit4,coef=12, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusHeartvsLung=topTable(fit4, coef=15, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusLungvsLiver=topTable(fit4, coef=18, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

## For a consistent FDR correction across contrasts, use decideTests in combination with method=&quot;global&quot; to adjust multiple testing across all genes and multiple related contrasts        

summary(decideTests(fit4, method=&quot;global&quot;, p.value=0.01))</code></pre>
<pre><code>   ChimpKidneyvsLiver HumanKidneyvsLiver RhesusKidneyvsLiver
-1               2408               2326                2804
0                7338               7530                6741
1                2438               2328                2639
   ChimpKidneyvsLung HumanKidneyvsLung RhesusKidneyvsLung
-1              2243              1788               2812
0               7532              8571               6381
1               2409              1825               2991
   ChimpKidneyvsHeart HumanKidneyvsHeart RhesusKidneyvsHeart
-1               2482               2264                3153
0                7274               7786                6166
1                2428               2134                2865
   ChimpHeartvsLiver HumanHeartvsLiver RhesusHeartvsLiver ChimpHeartvsLung
-1              3169              2543               3512             2666
0               5783              7047               5048             6912
1               3232              2594               3624             2606
   HumanHeartvsLung RhesusHeartvsLung ChimpLungvsLiver HumanLungvsLiver
-1             1902              3312             3140             2349
0              8272              5423             5943             7539
1              2010              3449             3101             2296
   RhesusLungvsLiver
-1              3683
0               5058
1               3443</code></pre>
<pre class="r"><code>colnames(decideTests(fit4, method=&quot;global&quot;, p.value=0.01))</code></pre>
<pre><code> [1] &quot;ChimpKidneyvsLiver&quot;  &quot;HumanKidneyvsLiver&quot;  &quot;RhesusKidneyvsLiver&quot;
 [4] &quot;ChimpKidneyvsLung&quot;   &quot;HumanKidneyvsLung&quot;   &quot;RhesusKidneyvsLung&quot; 
 [7] &quot;ChimpKidneyvsHeart&quot;  &quot;HumanKidneyvsHeart&quot;  &quot;RhesusKidneyvsHeart&quot;
[10] &quot;ChimpHeartvsLiver&quot;   &quot;HumanHeartvsLiver&quot;   &quot;RhesusHeartvsLiver&quot; 
[13] &quot;ChimpHeartvsLung&quot;    &quot;HumanHeartvsLung&quot;    &quot;RhesusHeartvsLung&quot;  
[16] &quot;ChimpLungvsLiver&quot;    &quot;HumanLungvsLiver&quot;    &quot;RhesusLungvsLiver&quot;  </code></pre>
<pre class="r"><code># Make the Venn diagrams

for(i in 1:6){
  title &lt;- sub(&quot;Chimp&quot;, &quot;&quot;, names(top4)[i])
  title &lt;- sub(&quot;vs&quot;, &quot; vs. &quot;, title)
  
  mylist &lt;- list()
  mylist[[&quot;Chimpanzee&quot;]] &lt;- row.names(top4[[names(top4)[i]]])[top4[[names(top4)[i]]]$adj.P.Val &lt; 0.01]
  mylist[[&quot;Human&quot;]] &lt;- row.names(top4[[names(top4)[i+6]]])[top4[[names(top4)[i+6]]]$adj.P.Val &lt; 0.01]
  mylist[[&quot;Rhesus macaque&quot;]] &lt;- row.names(top4[[names(top4)[i+12]]])[top4[[names(top4)[i+12]]]$adj.P.Val &lt; 0.01]

venn.diagram(mylist, filename = paste0(sub(&quot;Chimp&quot;, &quot;&quot;, names(top4)[i]), &quot;.png&quot;), imagetype = &quot;png&quot;, main=title, fill=pal[1:3], alpha=c(0.5, 0.5, 0.5), cex=1.5 , lty=1, height=2000, width=2000);                                                   

}

rl &lt;- lapply(list(&quot;HeartvsLiver.png&quot;, &quot;HeartvsLung.png&quot;, &quot;KidneyvsHeart.png&quot;, &quot;KidneyvsLiver.png&quot;, &quot;KidneyvsLung.png&quot;, &quot;LungvsLiver.png&quot;), png::readPNG)
gl &lt;- lapply(rl, grid::rasterGrob)
do.call(gridExtra::grid.arrange, gl)</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/pairwise%20tissues-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-between-2-species-across-tissues" class="section level4">
<h4>Differential expression between 2 species (across tissues)</h4>
<pre class="r"><code># Make contrast matrix

cm2 &lt;- makeContrasts(
                     ## pairwise species contrasts in each tissue                                                                                                                          
                     KidneyChimpvsHuman = Chimp.kidney-Human.kidney,
                     KidneyChimpvsRhesus = Chimp.kidney-Rhesus.kidney,
                     KidneyHumanvsRhesus = Human.kidney-Rhesus.kidney,
                     HeartChimpvsHuman = Chimp.heart-Human.heart,
                     HeartChimpvsRhesus = Chimp.heart-Rhesus.heart,
                     HeartHumanvsRhesus = Human.heart-Rhesus.heart,
                     LiverChimpvsHuman = Chimp.liver-Human.liver,
                     LiverChimpvsRhesus = Chimp.liver-Rhesus.liver,
                     LiverHumanvsRhesus = Human.liver-Rhesus.liver,
                     LungChimpvsHuman = Chimp.lung-Human.lung,
                     LungChimpvsRhesus = Chimp.lung-Rhesus.lung,
                     LungHumanvsRhesus = Human.lung-Rhesus.lung,
                     levels=design
                     )

# Linear model
fit2 &lt;- contrasts.fit(fit.cpm.voom.cyclic.ran, cm2)
fit2 &lt;- eBayes(fit2)

# Order by BH values
top2 &lt;- list(ChimpvsHuman_Kidney=topTable(fit2, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpvsRhesus_Kidney=topTable(fit2, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanvsRhesus_Kidney=topTable(fit2, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpvsHuman_Heart=topTable(fit2, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpvsRhesus_Heart=topTable(fit2, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanvsRhesus_Heart=topTable(fit2, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpvsHuman_Liver=topTable(fit2, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpvsRhesus_Liver=topTable(fit2, coef=8, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanvsRhesus_Liver=topTable(fit2, coef=9, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpvsHuman_Lung=topTable(fit2, coef=10, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpvsRhesus_Lung=topTable(fit2, coef=11, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanvsRhesus_Lung=topTable(fit2, coef=12, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Adjust FDR
summary(decideTests(fit2, method=&quot;global&quot;, p.value=0.01))</code></pre>
<pre><code>   KidneyChimpvsHuman KidneyChimpvsRhesus KidneyHumanvsRhesus
-1                670                1393                1324
0               10937                9262                9486
1                 577                1529                1374
   HeartChimpvsHuman HeartChimpvsRhesus HeartHumanvsRhesus
-1              1224               1754               2221
0               9702               8803               8054
1               1258               1627               1909
   LiverChimpvsHuman LiverChimpvsRhesus LiverHumanvsRhesus
-1              1030               1689               1628
0              10230               8758               8884
1                924               1737               1672
   LungChimpvsHuman LungChimpvsRhesus LungHumanvsRhesus
-1              598              1117              1425
0             11118              9925              9301
1               468              1142              1458</code></pre>
<pre class="r"><code>colnames(decideTests(fit2, method=&quot;global&quot;, p.value=0.01))</code></pre>
<pre><code> [1] &quot;KidneyChimpvsHuman&quot;  &quot;KidneyChimpvsRhesus&quot; &quot;KidneyHumanvsRhesus&quot;
 [4] &quot;HeartChimpvsHuman&quot;   &quot;HeartChimpvsRhesus&quot;  &quot;HeartHumanvsRhesus&quot; 
 [7] &quot;LiverChimpvsHuman&quot;   &quot;LiverChimpvsRhesus&quot;  &quot;LiverHumanvsRhesus&quot; 
[10] &quot;LungChimpvsHuman&quot;    &quot;LungChimpvsRhesus&quot;   &quot;LungHumanvsRhesus&quot;  </code></pre>
<pre class="r"><code># Make the Venn diagrams

# Make the labels

for(i in 1:3){
  # title &lt;- sub(&quot;Kidney&quot;, &quot;&quot;,top2_names[i])
  title &lt;- sub(&quot;_Kidney&quot;, &quot;&quot;, names(top2)[i])
  title &lt;- sub(&quot;vs&quot;, &quot; vs. &quot;, title)
  
  mylist &lt;- list()
  mylist[[&quot;Kidney&quot;]] &lt;- row.names(top2[[names(top2)[i]]])[top2[[names(top2)[i]]]$adj.P.Val &lt; 0.01]
  mylist[[&quot;Liver&quot;]] &lt;- row.names(top2[[names(top2)[i+3]]])[top2[[names(top2)[i+3]]]$adj.P.Val &lt; 0.01]
  mylist[[&quot;Lung&quot;]] &lt;- row.names(top2[[names(top2)[i+6]]])[top2[[names(top2)[i+6]]]$adj.P.Val &lt; 0.01]
  mylist[[&quot;Heart&quot;]] &lt;- row.names(top2[[names(top2)[i+9]]])[top2[[names(top2)[i+9]]]$adj.P.Val &lt; 0.01]

venn.diagram(mylist, filename = paste0(sub(&quot;_Kidney&quot;, &quot;&quot;, names(top2)[i]), &quot;.png&quot;), imagetype = &quot;png&quot;, main=title, fill = pal[1:4], alpha=c(0.5, 0.5, 0.5, 0.5), cex=1.5 , lty=1, height=2000, width=2000);                                               

}

rl &lt;- lapply(list(&quot;Chimpanzee vs Human.png&quot;, &quot;Chimpanzee vs Rhesus macaque.png&quot;, &quot;Human vs Rhesus macaque.png&quot;), png::readPNG)
gl &lt;- lapply(rl, grid::rasterGrob)
do.call(gridExtra::grid.arrange, gl)</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/pairwise%20species-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-between-3-species-in-the-same-tissue" class="section level4">
<h4>Differential expression between 3 species (in the same tissue)</h4>
<pre class="r"><code>for(i in 1:4){
  title &lt;- sub(&quot;ChimpvsHuman_&quot;, &quot;&quot;, names(top2)[(i*3)-2])

  mylist &lt;- list()
  mylist[[&quot;Chimp vs. Human&quot;]] &lt;- row.names(top2[[names(top2)[(i*3)-2]]])[top2[[names(top2)[(i*3)-2]]]$adj.P.Val &lt; 0.01]
  mylist[[&quot;Chimp vs. Rhesus&quot;]] &lt;- row.names(top2[[names(top2)[(i*3)-1]]])[top2[[names(top2)[(i*3)-1]]]$adj.P.Val &lt; 0.01]
  mylist[[&quot;Human vs. Rhesus&quot;]] &lt;- row.names(top2[[names(top2)[(i*3)]]])[top2[[names(top2)[(i*3)]]]$adj.P.Val &lt; 0.01]

venn.diagram(mylist, filename = paste0(title, &quot;_all_species.png&quot;), imagetype = &quot;png&quot;, main=title, fill=pal[1:3], alpha=c(0.5, 0.5, 0.5), cex=1.5 , lty=1, height=2000, width=2000, margin=0.1, cat.dist=0.1);
}

rl &lt;- lapply(list(&quot;Kidney_all_species.png&quot;, &quot;Liver_all_species.png&quot;, &quot;Lung_all_species.png&quot;, &quot;Heart_all_species.png&quot;), png::readPNG)
gl &lt;- lapply(rl, grid::rasterGrob)
do.call(gridExtra::grid.arrange, gl)</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/single%20tissue%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>To do: look more at directionality</strong> A gene could be DE between humans and chimps, upregulated in chimps as well as DE between humans and rhesus but be downregulated in rhesus.</p>
</div>
</div>
<div id="part-2-tissue-specific-expression-in-and-across-species" class="section level3">
<h3>Part 2: Tissue specific expression in and across species</h3>
<pre class="r"><code># Make the contrast matrix
cm3 &lt;- makeContrasts(
                     ChimpKidneyvsOthers = Chimp.kidney-(Chimp.liver+Chimp.heart+Chimp.lung)/3,
                     HumanKidneyvsOthers = Human.kidney-(Human.liver+Human.heart+Human.lung)/3,
                     RhesusKidneyvsOthers = Rhesus.kidney-(Rhesus.liver+Rhesus.heart+Rhesus.lung)/3,
                     ChimpLivervsOthers = Chimp.liver-(Chimp.kidney+Chimp.heart+Chimp.lung)/3,
                     HumanLivervsOthers = Human.liver-(Human.kidney+Human.heart+Human.lung)/3,
                     RhesusLivervsOthers = Rhesus.liver-(Rhesus.kidney+Rhesus.heart+Rhesus.lung)/3,
                     ChimpHeartvsOthers = Chimp.heart-(Chimp.liver+Chimp.kidney+Chimp.lung)/3,
                     HumanHeartvsOthers = Human.heart-(Human.liver+Human.kidney+Human.lung)/3,
                     RhesusHeartvsOthers = Rhesus.heart-(Rhesus.liver+Rhesus.kidney+Rhesus.lung)/3,
                     ChimpLungvsOthers = Chimp.lung-(Chimp.liver+Chimp.heart+Chimp.kidney)/3,
                     HumanLungvsOthers = Human.lung-(Human.liver+Human.heart+Human.kidney)/3,
                     RhesusLungvsOthers = Rhesus.lung-(Rhesus.liver+Rhesus.heart+Rhesus.kidney)/3,
                     levels=design
                     )
fit3 &lt;- contrasts.fit(fit.cpm.voom.cyclic.ran, cm3)
fit3 &lt;- eBayes(fit3)

# Find the top hits by BH adjusted p-value
top3 &lt;- list(ChimpKidney=topTable(fit3, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpLiver=topTable(fit3, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpHeart=topTable(fit3, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), ChimpLung=topTable(fit3, coef=10, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanKidney=topTable(fit3, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanLiver=topTable(fit3, coef=5, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanHeart=topTable(fit3, coef=8, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HumanLung=topTable(fit3, coef=11, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusKidney=topTable(fit3, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusLiver=topTable(fit3, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusHeart=topTable(fit3, coef=9, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), RhesusLung=topTable(fit3, coef=12, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Find number of DE genes at FDR cutoff &lt; 1% for each tissue in all 3 species

summary(top3$ChimpKidney$adj.P.Val &lt; 0.01 &amp; top3$HumanKidney$adj.P.Val &lt; 0.01 &amp; top3$RhesusKidney$adj.P.Val &lt; 0.01) # 1970                                                              </code></pre>
<pre><code>   Mode   FALSE    TRUE    NA&#39;s 
logical   10214    1970       0 </code></pre>
<pre class="r"><code>summary(top3$ChimpHeart$adj.P.Val &lt; 0.01 &amp; top3$HumanHeart$adj.P.Val &lt; 0.01 &amp; top3$RhesusHeart$adj.P.Val &lt; 0.01) # 2738                                                                </code></pre>
<pre><code>   Mode   FALSE    TRUE    NA&#39;s 
logical    9446    2738       0 </code></pre>
<pre class="r"><code>summary(top3$ChimpLiver$adj.P.Val &lt; 0.01 &amp; top3$HumanLiver$adj.P.Val &lt; 0.01 &amp; top3$RhesusLiver$adj.P.Val &lt; 0.01) # 3502                                                         </code></pre>
<pre><code>   Mode   FALSE    TRUE    NA&#39;s 
logical    8682    3502       0 </code></pre>
<pre class="r"><code>summary(top3$ChimpLung$adj.P.Val &lt; 0.01 &amp; top3$HumanLung$adj.P.Val &lt; 0.01 &amp; top3$RhesusLung$adj.P.Val &lt; 0.01) # 2610</code></pre>
<pre><code>   Mode   FALSE    TRUE    NA&#39;s 
logical    9574    2610       0 </code></pre>
<pre class="r"><code># Make the Venn diagrams

# Make the labels

#for(i in 1:4){
# title &lt;- sub(&quot;Kidney&quot;, &quot;&quot;,top2_names[i])
#  title &lt;- sub(&quot;Chimp&quot;, &quot;&quot;, names(top3)[i])
#  title &lt;- sub(&quot;vs&quot;, &quot; vs. &quot;, title)
  
#  mylist &lt;- list()
#  mylist[[&quot;Kidney&quot;]] &lt;- row.names(top3[[names(top3)[i]]])[top3[[names(top3)[i]]]$adj.P.Val &lt; 0.01]
#  mylist[[&quot;Liver&quot;]] &lt;- #row.names(top3[[names(top3)[i+3]]])[top3[[names(top3)[i+3]]]$adj.P.Val &lt; 0.01]
#  mylist[[&quot;Lung&quot;]] &lt;- row.names(top3[[names(top3)[i+6]]])[top3[[names(top3)[i+6]]]$adj.P.Val &lt; 0.01]
#  mylist[[&quot;Heart&quot;]] &lt;- #row.names(top3[[names(top3)[i+9]]])[top3[[names(top3)[i+9]]]$adj.P.Val &lt; 0.01]

#venn.diagram(mylist, filename = paste0(sub(&quot;Chimp&quot;, &quot;&quot;, names(top3)[i]), &quot;.png&quot;), imagetype = &quot;png&quot;, main=title, fill = pal[1:4], alpha=c(0.5, 0.5, 0.5, 0.5), cex=1.5 , lty=1, height=2000, width=2000);                                               

#}

#rl &lt;- lapply(list(&quot;Kidney.png&quot;, &quot;Liver.png&quot;, &quot;Lung.png&quot;, &quot;Heart.png&quot;), png::readPNG)
#gl &lt;- lapply(rl, grid::rasterGrob)
#do.call(gridExtra::grid.arrange, gl)

Tissues &lt;- c(&quot;Heart&quot;, &quot;Kidney&quot;, &quot;Liver&quot;, &quot;Lung&quot;)

for(i in 1:4){
  title &lt;- paste(Tissues[i], &quot;specific DE genes&quot;, sep=&quot; &quot;)
  
  mylist &lt;- list()
  mylist[[&quot;Chimpanzee&quot;]] &lt;- row.names(top3[[names(top3)[i]]])[top3[[names(top3)[i]]]$adj.P.Val &lt; 0.01] 
  mylist[[&quot;Human&quot;]] &lt;-  row.names(top3[[names(top3)[i+4]]])[top3[[names(top3)[i+4]]]$adj.P.Val &lt; 0.01]      
  mylist[[&quot;Rhesus Macaque&quot;]] &lt;-  row.names(top3[[names(top3)[i+8]]])[top3[[names(top3)[i+8]]]$adj.P.Val &lt; 0.01]       

  venn.diagram(mylist, filename = paste(Tissues[i], &quot;specific DE genes.png&quot;, sep=&quot; &quot;), imagetype = &quot;png&quot;, main=title, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=2000)
}

# View the Venn Diagram

rl &lt;- lapply(list(&quot;Heart specific DE Genes.png&quot;, &quot;Kidney specific DE Genes.png&quot;, &quot;Liver specific DE Genes.png&quot;, &quot;Lung specific DE Genes.png&quot;), png::readPNG)
gl &lt;- lapply(rl, grid::rasterGrob)
do.call(gridExtra::grid.arrange, gl)</code></pre>
<p><img src="figure/DE_genes_pairwise_ts_specific.Rmd/tissue%20specific%20exp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Next to do: Variance analysis for the 3 tissues that the one tissue was different from. We are interested in situations where there is a big difference between one tissue and the other three but only a small variance between the three to count as truly “tissue specific”.</p>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
