---
title: "Expression_methylation_visualization"
author: "Lauren Blake"
date: "June 1, 2017"
output: html_document
---

# We want to make a dropdown menu with gene expression data from the tissue project

```{r, echo = FALSE}
options(warn=-1)
# Install the package from GitHub
# devtools::install_github("walkerke/bsselectR")

# Call the library
library(bsselectR)
library(ggplot2)
#library('rsconnect')
#library('shiny')


```


```{r, echo = FALSE}
# Import the tissue data

samples <- read.delim("../data/Sample_info_RNAseq_limma.txt")

# Eliminate H1H
samples <- samples[-17,]
#dim(samples)

# Make labels

species <- samples$Species
tissue <- samples$Tissue
labels <- paste(samples$Species, samples$Tissue, sep=" ")

# Get the data

genes_in_methyl <- read.table("/Users/laurenblake/Dropbox/primate_BS-seq_project/Expression_Methylation_Overlap/genes_in_methyl_2345.txt")

#dim(genes_in_methyl)

methyl_data <- genes_in_methyl[1:2,]


genes_in_exp <- read.table("/Users/laurenblake/Dropbox/primate_BS-seq_project/Expression_Methylation_Overlap/genes_in_exp_2345.txt")

#dim(genes_in_exp)

normalized_data <- genes_in_exp[1:2,]

```

```{r, echo = FALSE}

# Needed for pwalk command later
library(purrr)

# Needed for str_replace_all command

library(stringr)
library(bsselectR)
library("ggplot2")
library("gridExtra")

colnames(normalized_data) <- labels

# Make plots with color-blind friendl(ier) colors


# Expression
plots <- lapply(rownames(normalized_data), function(x) {
  gene_expression = as.data.frame(normalized_data[grepl(x, rownames(normalized_data)), ])
  gene_expression_t <- as.data.frame(t(gene_expression))
  df <- as.data.frame(cbind(gene_expression_t, labels, tissue))
  colnames(df) <- c("Expression", "Labels", "Tissue")
  p <- ggplot(df, aes(x = factor(df[,2]), y = df[,1], fill = factor(Tissue))) + geom_boxplot() + theme_bw() + theme(axis.text.x=element_text(angle = 90, hjust = 0)) + xlab("Species-tissue pair") + ylab("Normalized expression") + ggtitle("Expression ", paste(x))   + scale_fill_discrete(name="Tissue Type",  labels = c("Heart", "Kidney", "Liver", "Lung"))                      
   
  
  })

paths <- paste0(rownames(normalized_data), ".pdf")
pwalk(list(paths, plots), ggsave, path = "test_interact_plots_exp", dpi = 300)
  
# Methylation

plots <- lapply(rownames(normalized_data), function(x) {
  methylation = as.data.frame(methyl_data[grepl(x, rownames(methyl_data)), ])
  methylation_t <- as.data.frame(t(methylation))
  df <- as.data.frame(cbind(methylation_t, labels, tissue))
  colnames(df) <- c("Methylation",  "Labels", "Tissue")
  p <- ggplot(df, aes(x = factor(df[,2]), y = df[,1], fill = factor(Tissue))) + geom_boxplot() + theme_bw() + theme(axis.text.x=element_text(angle = 90, hjust = 0)) + xlab("Species-tissue pair") + ylab("Average of Smoothed Methylation Level") + ggtitle("Methylation ", paste(x))   + scale_fill_discrete(name="Tissue Type",  labels = c("Heart", "Kidney", "Liver", "Lung"))    
   
  
  })


paths <- paste0(rownames(normalized_data), ".pdf")
pwalk(list(paths, plots), ggsave, path = "test_interact_plots_methyl", dpi = 300)
 





gene_expression_plots <- paste0(list.files("test_interact_plots", full.names = TRUE))


names(gene_expression_plots) <- str_replace_all(gene_expression_plots, c("\\.png" = "", "test_interact_plots/" = ""))


bsselect(gene_expression_plots, type = "img",  live_search = TRUE, show_tick = TRUE)


                  
  
```

