---
title: "Integrate_methyl_gene_expression"
author: "Lauren Blake"
date: "September 19, 2017"
output: html_document
---

In this script, we will perform a joint analysis of methylation and gene expression levels. We will do this using the framework from Pai et al. 2011, "A Genome-Wide Study of DNA Methylation Patterns and Gene Expression Levels in Multiple Human and Chimpanzee Tissues". Here is some of the relevant text:

To examine whether changes in gene expression levels between humans and chimpanzees (within each tissue) can be explained by inter-species differences in methylation levels, we extended the linear mixed-effects model framework described in Blekhman et al. (2008) to include methylation as a covariate. However, since we have to correct the multi-species array data for probe-effects, it is difficult to interpret the methylation coefficient when it is added directly to the model, since it is confounded with the probe effects. Consequently, we used an alternative approach in which we used regression to correct for the methylation effect. Specifically, for each gene-tissue combination, we tested for differences in expression level between human and chimpanzee after regressing out the following effects:

Expression microarray probe effects only
Expression microarray probe effects and CpG-specific methylation levels

This script shows that, for a given pair (e.g. human versus chimpanzee liver) we should use the genes DE between human and chimpanzee liver only rather than all genes. The method using only DE genes is what Pai et al. used, but it was not entirely clear in the main paper.   

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```


# Load and normalize data

```{r}
# Load library

library("lme4")
library("edgeR")
library("limma")
library("dplyr")
library("plyr")

# Load average methylation data

methyl_values <- read.csv("../data/3965_avg_methyl_per_ts_gene.txt", sep="", stringsAsFactors=FALSE)

# For now, take out samples with combined regions
one_gene_rows <- !grepl(",", methyl_values$X)

methyl_values_one_gene <-  cbind(methyl_values, one_gene_rows)

methyl_values_for_exp <- subset(methyl_values_one_gene, one_gene_rows == "TRUE")
methyl_values_for_exp <- methyl_values_for_exp[,1:49]

# Get ENSG-gene pairings

ENSG_GENE_HG19 <- read.csv("../data/ENSG_GENE_HG19.csv", stringsAsFactors=FALSE, header = T)

# Note: we eliminated H1H in the expression data, so we are going to do this with the methylation data for this analysis

methyl_values_for_exp_no_H1H <- methyl_values_for_exp[,-17]

methyl_values_for_exp_no_H1H_ENSG <- merge(methyl_values_for_exp_no_H1H, ENSG_GENE_HG19, by.x = "X", by.y = "Gene")

# Load sample data

samples <- read.delim("../data/Sample_info_RNAseq_limma.txt")

# Eliminate H1H
samples <- samples[-17,]
dim(samples)

# Make labels

labels <- paste(samples$Species, samples$Tissue, sep=" ")

## Make the contrast matrix 

species <- samples$Species
tissue <- samples$Tissue


# Retrieve RIN score for each sample
RNA_seq_info <- read.csv("../data/RNA_seq_info.csv")
RIN <- as.data.frame(RNA_seq_info[,22])
RIN <- as.matrix(RIN)
colnames(RIN) <- c("RIN")

## Make the contrast matrix and rename columns of the contrast matrix

design <- model.matrix(~ 0 + labels + RIN)

# Load expression data

# Load count data

counts_genes_in_cutoff <- read.delim("../data/counts_12184.txt")

# TMM 

dge_in_cutoff <- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff <- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff <- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)
head(cpm_in_cutoff)
hist(cpm_in_cutoff, xlab = "Log2(CPM)", main = "Log2(CPM) values for genes meeting the filtering criteria", breaks = 100 )

# Voom with individual as a random variable

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=T)

#corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit.consensus <- 0.2197275

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit.consensus)

# Get the expression values for the genes that we have average methylation data

methyl_genes_in_exp <- rownames(cpm.voom.cyclic$E) %in% methyl_values_for_exp_no_H1H_ENSG$ensg
methyl_genes_in_exp <- as.data.frame(methyl_genes_in_exp)
summary(methyl_genes_in_exp )

counts_genes_in <- cbind(cpm.voom.cyclic$E, methyl_genes_in_exp)
counts_genes_in_cutoff <- subset(counts_genes_in, methyl_genes_in_exp == "TRUE")
exp_levels_3776 <- counts_genes_in_cutoff[,1:47]

# Make ensg names the last column

exp_levels_3776[,48] <- rownames(exp_levels_3776)

# Merge expression and methylation

exp_methyl <- merge(exp_levels_3776, methyl_values_for_exp_no_H1H_ENSG, by.x = "V48", by.y = "ensg")
rownames(exp_methyl) <- exp_methyl$V48

# Separate expression and methylation

expression_values_only <- exp_methyl[,2:48]
methylation_values_only <- exp_methyl[,50:96]
```

# Run H-C liver using all genes (not just the DE genes)

## Write the linear models for scenarios 1 and 2

Scenario 1: don't need to regress out anything

gene expression = species + RIN 

Scenario 2: going to regress out methylation

residuals = species + RIN 

## Obtain corrected data (regression for scenario 2)

```{r}
human_chimp_liver_samples <- c(3, 7, 11, 15, 18, 22, 26, 30)

# Make an array to hold the residuals

resid_methyl <- array(0, dim = c(3776, 8)) 

# Regress out RIN on a gene-by-gene basis
expression_values_only <- as.data.frame(expression_values_only)
methylation_values_only <- as.data.frame(methylation_values_only)


for (i in 1:3776){
    resid_methyl[i,] <- lm(t(expression_values_only[i,human_chimp_liver_samples]) ~ t(methylation_values_only[i,human_chimp_liver_samples]))$resid
}

rownames(resid_methyl) <- rownames(expression_values_only)

```

## Run the linear model using the corrected data (start with H v C Liver example)

Note: when you use the LRT, you can only do pairwise comparisons so compare 2 species (without the 3rd) or group species HC v R or H v RC

```{r}
# Select data- human versus chimp liver

limma.ash <- readRDS("../data/results-limma-voom-ash.rds", refhook = NULL)
human_chimp_liver <- limma.ash$HvC_Liver

# Pick the ENSG

human_chimp_liver_3965 <- human_chimp_liver$genes %in% rownames(expression_values_only)
human_chimp_liver_3965 <- as.data.frame(human_chimp_liver_3965)
counts_genes_in <- cbind(human_chimp_liver, human_chimp_liver_3965)
counts_genes_in_cutoff <- subset(counts_genes_in, methyl_genes_in_exp == "TRUE")
  
human_chimp_liver_de_limma <- counts_genes_in_cutoff[,1:14]

# Get the values only for the human and chimp liver

human_chimp_liver_samples <- c(3, 7, 11, 15, 18, 22, 26, 30)
species_no_rhesus <- species[human_chimp_liver_samples]
species_no_rhesus <- droplevels.factor(species_no_rhesus, "Rhesus")

# Make the design matrix
design <- model.matrix(~ 0 + as.factor(species_no_rhesus) + RIN[human_chimp_liver_samples,])

# Scenario 1

fit1 <- lmFit(expression_values_only[,human_chimp_liver_samples], design)
fit1 <- eBayes(fit1)
HvCLiver_fit1 = topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")

# Scenario 2

fit2 <- lmFit(resid_methyl, design)
fit2 <- eBayes(fit2)
HvCLiver_fit2 = topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")

# Bind the DE initially, DE from fit1 and DE from fit2 together

HvCLiver_fits12 <- as.data.frame(cbind(rownames(HvCLiver_fit1), HvCLiver_fit1$adj.P.Val, HvCLiver_fit2$adj.P.Val), stringsAsFactors = FALSE)
HvCLiver_fits12[,2] <- as.numeric(HvCLiver_fits12[,2])
HvCLiver_fits12[,3] <- as.numeric(HvCLiver_fits12[,3])
colnames(HvCLiver_fits12) <- c("genes", "fit1_FDR", "fit2_FDR")

HvCLiver_DE_fits12 <- merge(human_chimp_liver_de_limma, HvCLiver_fits12, by = "genes")

# Set FDR level

FDR_level <- 0.05

DE_total <- HvCLiver_DE_fits12[which(HvCLiver_DE_fits12$adj.P.Val < FDR_level),]
nrow(DE_total)

DE_always <- HvCLiver_DE_fits12[which(HvCLiver_DE_fits12$adj.P.Val < FDR_level & HvCLiver_DE_fits12$fit1_FDR < FDR_level &  HvCLiver_DE_fits12$fit2_FDR < FDR_level),]
nrow(DE_always)

DE_before <- HvCLiver_DE_fits12[which(HvCLiver_DE_fits12$adj.P.Val > FDR_level & HvCLiver_DE_fits12$fit1_FDR < FDR_level &  HvCLiver_DE_fits12$fit2_FDR > FDR_level),]
nrow(DE_before)

DE_after <- HvCLiver_DE_fits12[which(HvCLiver_DE_fits12$adj.P.Val > FDR_level & HvCLiver_DE_fits12$fit1_FDR > FDR_level &  HvCLiver_DE_fits12$fit2_FDR < FDR_level),]
nrow(DE_after)

summary(HvCLiver_DE_fits12$fit2_FDR)



DE_before <- HvCLiver_DE_fits12[which(HvCLiver_DE_fits12$fit1_FDR < FDR_level &  HvCLiver_DE_fits12$fit2_FDR > FDR_level),]
nrow(DE_before)

DE_after <- HvCLiver_DE_fits12[which(HvCLiver_DE_fits12$fit1_FDR > FDR_level &  HvCLiver_DE_fits12$fit2_FDR < FDR_level),]
nrow(DE_after)
```

### Permute the methylation values, then re-run

```{r}
# Run 1 for all genes
run1 <- array(0, dim = c(3776, 8)) 

for (i in 1:3776){
  df <- as.data.frame(methylation_values_only[i,human_chimp_liver_samples])
  new_df <- as.data.frame(sample(df))
  run1[i,] <- t(new_df)
}



# Make an array to hold the residuals

resid_methyl_perm <- array(0, dim = c(3776, 8)) 

# Regress out RIN on a gene-by-gene basis
expression_values_only <- as.data.frame(expression_values_only)
methylation_values_only <- as.data.frame(methylation_values_only)


for (i in 1:3776){
    resid_methyl_perm[i,] <- lm(t(expression_values_only[i,human_chimp_liver_samples]) ~ run1[i,])$resid
}

rownames(resid_methyl_perm) <- rownames(expression_values_only)


# Scenario 2

fit2 <- lmFit(resid_methyl_perm, design)
fit2 <- eBayes(fit2)
HvCLiver_perm = topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")

```










# START WITH ONLY H-C LIVER DE GENES

## Obtain corrected data (regression for scenarios 1 and 2)

```{r}
# Select FDR level

FDR_level <- 0.05

# Select data- DE at given FDR for human versus chimp liver

human_chimp_liver_DE <- human_chimp_liver[which(human_chimp_liver$adj.P.Val < FDR_level), ]
dim(human_chimp_liver_DE)

# Get expression and methylation levels only for DE genes

exp_methyl_DE <- exp_methyl$V48 %in% human_chimp_liver_DE$genes
inshared_lists_data <- as.data.frame(exp_methyl_DE)
counts_genes_in <- cbind(exp_methyl, inshared_lists_data)
counts_genes_in_cutoff <- subset(counts_genes_in, inshared_lists_data == "TRUE")
exp_methyl_DE_only <- counts_genes_in_cutoff[,1:96]


#cpm.voom.cyclic$E 

#exp_methyl_DE <- rownames(cpm.voom.cyclic$E)  %in% human_chimp_liver_DE$genes
#inshared_lists_data <- as.data.frame(exp_methyl_DE)
#counts_genes_in <- cbind(cpm.voom.cyclic$E, inshared_lists_data)
#counts_genes_in_cutoff <- subset(counts_genes_in, inshared_lists_data == "TRUE")
#exp_DE_only <- counts_genes_in_cutoff[,1:47]


# Separate expression and methylation

expression_values_only <- exp_methyl_DE_only[,2:48]
methylation_values_only <- exp_methyl_DE_only[,50:96]

# Get the values only for the human and chimp liver

human_chimp_liver_samples <- c(3, 7, 11, 15, 18, 22, 26, 30)
species_no_rhesus <- species[human_chimp_liver_samples]
species_no_rhesus <- droplevels.factor(species_no_rhesus, "Rhesus")

tissue_no_rhesus <- tissue[human_chimp_liver_samples]

# Make the design matrix
design <- model.matrix(~ as.factor(species_no_rhesus) + RIN[human_chimp_liver_samples,])


# Make an array to hold the residuals

resid_methyl <- array(0, dim = c(nrow(expression_values_only), 8)) 

# Regress out RIN on a gene-by-gene basis
expression_values_only <- as.data.frame(expression_values_only)
methylation_values_only <- as.data.frame(methylation_values_only)


for (i in 1:nrow(expression_values_only)){
    resid_methyl[i,] <- lm(t(expression_values_only[i,human_chimp_liver_samples]) ~ t(methylation_values_only[i,human_chimp_liver_samples]))$resid
}

rownames(resid_methyl) <- rownames(expression_values_only)

```

## Run the linear model using the corrected data (start with H v C Liver example)

Note: when you use the LRT, you can only do pairwise comparisons so compare 2 species (without the 3rd) or group species HC v R or H v RC

```{r}

# Scenario 1

fit1 <- lmFit(cpm.voom.cyclic[,human_chimp_liver_samples], design)
fit1 <- eBayes(fit1)
HvCLiver_fit1 = topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")

use_genes <- HvCLiver_fit1[which(HvCLiver_fit1$adj.P.Val < 0.1),] 

exp_methyl_DE <- exp_methyl$V48 %in% use_genes$genes
inshared_lists_data <- as.data.frame(exp_methyl_DE)
counts_genes_in <- cbind(exp_methyl, inshared_lists_data)
counts_genes_in_cutoff <- subset(counts_genes_in, inshared_lists_data == "TRUE")
exp_DE_only <- counts_genes_in_cutoff[,1:96]


expression_values_only <- exp_DE_only[,2:48]
methylation_values_only <- exp_DE_only[,50:96]



fit1 <- lmFit(expression_values_only[,human_chimp_liver_samples], design)
fit1 <- eBayes(fit1)
HvCLiver_fit3 = topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")

summary(HvCLiver_fit3$adj.P.Val < 0.1)

HvCLiver_fit_first_model <- HvCLiver_fit3[which(HvCLiver_fit3$adj.P.Val < 0.1), ]

# Scenario 2

exp_methyl_DE <- exp_methyl$V48 %in% rownames(HvCLiver_fit_first_model)
inshared_lists_data <- as.data.frame(exp_methyl_DE)
counts_genes_in <- cbind(exp_methyl, inshared_lists_data)
counts_genes_in_cutoff <- subset(counts_genes_in, inshared_lists_data == "TRUE")
exp_methyl_DE_only <- counts_genes_in_cutoff[,1:96]

expression_values_only <- exp_methyl_DE_only[,2:48]
methylation_values_only <- exp_methyl_DE_only[,50:96]

# Make an array to hold the residuals

resid_methyl <- array(0, dim = c(nrow(expression_values_only), 8)) 

# Regress out RIN on a gene-by-gene basis
expression_values_only <- as.data.frame(expression_values_only)
methylation_values_only <- as.data.frame(methylation_values_only)


for (i in 1:nrow(expression_values_only)){
    resid_methyl[i,] <- lm(t(expression_values_only[i,human_chimp_liver_samples]) ~ t(methylation_values_only[i,human_chimp_liver_samples]))$resid
}




fit2 <- lmFit(resid_methyl, design)
fit2 <- eBayes(fit2)
HvCLiver_fit2 = topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")

# Bind the DE initially, DE from fit1 and DE from fit2 together

HvCLiver_fits12 <- as.data.frame(cbind(rownames(HvCLiver_fit_first_model), HvCLiver_fit_first_model$adj.P.Val, HvCLiver_fit2$adj.P.Val), stringsAsFactors = FALSE)
HvCLiver_fits12[,2] <- as.numeric(HvCLiver_fits12[,2])
HvCLiver_fits12[,3] <- as.numeric(HvCLiver_fits12[,3])
colnames(HvCLiver_fits12) <- c("genes", "fit1_FDR", "fit2_FDR")

# Get numbers for actual data

DE_total <- HvCLiver_fits12[which(HvCLiver_fits12$fit1_FDR < FDR_level &  HvCLiver_fits12$fit2_FDR < FDR_level),]
nrow(DE_total)

DE_before <- HvCLiver_fits12[which(HvCLiver_fits12$fit1_FDR < FDR_level &  HvCLiver_fits12$fit2_FDR > FDR_level),]
nrow(DE_before)

DE_after <- HvCLiver_fits12[which(HvCLiver_fits12$fit1_FDR > FDR_level &  HvCLiver_fits12$fit2_FDR < FDR_level),]
nrow(DE_after)

summary(HvCLiver_fits12$fit2_FDR)
```

### Permute the methylation values, then re-run

```{r}
run1000 <- array(0, dim = c(3, 1000)) 

# Normally, we would permute 1000 times; in this example, we will permute 10 times
# for (k in 1:1000){

for (k in 1:10){
  
# Run 1 for all genes
run1 <- array(0, dim = c(nrow(expression_values_only), 8)) 

for (i in 1:nrow(expression_values_only)){
  df <- as.data.frame(methylation_values_only[i,human_chimp_liver_samples])
  new_df <- as.data.frame(sample(df))
  run1[i,] <- t(new_df)
}



# Make an array to hold the residuals

resid_methyl_perm <- array(0, dim = c(nrow(expression_values_only), 8)) 


#for (i in 1:2){
for (i in 1:nrow(expression_values_only)){
    resid_methyl_perm[i,] <- lm(t(expression_values_only[i,human_chimp_liver_samples]) ~ run1[i,])$resid
}

rownames(resid_methyl_perm) <- rownames(expression_values_only)


# Scenario 2

fit2 <- lmFit(resid_methyl_perm, design)
fit2 <- eBayes(fit2)
HvCLiver_perm = topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")

summary(HvCLiver_perm$adj.P.Val)

HvCLiver_DE_perm <- cbind(HvCLiver_fits12$fit1_FDR, HvCLiver_perm$adj.P.Val)

DE_total <- as.data.frame(HvCLiver_DE_perm[which(HvCLiver_DE_perm[,1] < FDR_level & HvCLiver_DE_perm[,2]  < FDR_level),])
DE_total_report <- nrow(DE_total)

DE_before <- as.data.frame(HvCLiver_DE_perm[which(HvCLiver_DE_perm[,1]  < FDR_level &  HvCLiver_DE_perm[,2]  > FDR_level),])
DE_before_report <- nrow(DE_before)

DE_after <- as.data.frame(HvCLiver_DE_perm[which(HvCLiver_DE_perm[,1] > FDR_level &  HvCLiver_DE_perm[,2] < FDR_level),])
DE_after_report <- nrow(DE_after)

new_numbers <- rbind(DE_total_report, DE_before_report, DE_after_report)

run1000[,k] <- new_numbers
}

median(run1000[1,1:10])
median(run1000[2,1:10])
median(run1000[3,1:10])

```











