---
title: "Integration_methyl_exp_human_chimp_only"
author: "Lauren Blake"
date: "October 4, 2017"
output: html_document
---

In this script, we will perform a joint analysis of methylation and gene expression levels in humans and chimpanzees only. We chose only these two species because the number of genes that we had methylation values for around the promoter was quite low when we required orthology between humans, chimps, and rhesus. 

# Load data


```{r}
# Load library

library("lme4")
library("edgeR")
library("limma")
library("dplyr")
library("plyr")

# Load average methylation data

methyl_values <- read.csv("../data/chimp_human_orth_7725_avg_methyl_per_ts_gene.txt", sep="", stringsAsFactors=FALSE)

# Note: we eliminated H1H in the expression data, so we are going to do this with the methylation data for this analysis

methyl_values_for_exp_no_H1H <- methyl_values[,-17]

# Load sample data

samples <- read.delim("../data/Sample_info_RNAseq_limma.txt")

# Eliminate H1H
samples <- samples[-17,]
dim(samples)

# Make labels

labels <- paste(samples$Species, samples$Tissue, sep=" ")

## Make the contrast matrix 

species <- samples$Species
tissue <- samples$Tissue


# Retrieve RIN score for each sample
RNA_seq_info <- read.csv("../data/RNA_seq_info.csv")
RIN <- as.data.frame(RNA_seq_info[,22])
RIN <- as.matrix(RIN)
colnames(RIN) <- c("RIN")

# Sample info needed

#new_sample_info <- cbind(samples[1:31,], RIN[1:31,])
#new_sample_info[,1] <- as.character(new_sample_info[,1])
#new_sample_info[,2] <- as.character(new_sample_info[,2])
#new_sample_info[,3] <- as.character(new_sample_info[,3])
#new_sample_info[,4] <- as.character(new_sample_info[,4])

write.table(new_sample_info, "../data/human_chimp_orth_new_sample_info.txt", quote = F)


## Make the contrast matrix and rename columns of the contrast matrix

design <- model.matrix(~ species + tissue + RIN)

# Load expression data

# Load count data

counts_genes_in_cutoff <- read.delim("../data/counts_12184.txt")

# TMM 

dge_in_cutoff <- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff <- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff <- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)
head(cpm_in_cutoff)
hist(cpm_in_cutoff, xlab = "Log2(CPM)", main = "Log2(CPM) values for genes meeting the filtering criteria", breaks = 100 )

# Voom with individual as a random variable

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=T)

corfit <- duplicateCorrelation(cpm.voom.cyclic, design, block=samples$Individual)
corfit.consensus <- 0.06659086

# Final voom on filtered data

cpm.voom.cyclic <- voom(dge_in_cutoff, design, normalize.method="cyclicloess", plot=TRUE, block=samples$Individual, correlation=corfit.consensus)

fit1 <- lmFit(cpm.voom.cyclic, design, block=samples$Individual, correlation=corfit$consensus)
fit1 <- eBayes(fit1)

HvC_full_model <- topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")
summary(HvC_full_model$logFC)

summary(HvC_full_model$logFC - HvC_Heart_fit_all$logFC)

plot(HvC_full_model$logFC, HvC_Heart_fit_all$logFC, xlab = "Full model- human and chimp hearts", ylab = "Model with human and chimp hearts only", main = "Beta for difference between chimp hearts and human hearts")
abline(a = 0, b = 1, col = "red")

plot(HvC_full_model$t, HvC_Heart_fit_all$t, xlab = "Full model- human and chimp hearts", ylab = "Model with human and chimp hearts only", main = "T for the beta")
abline(a = 0, b = 1, col = "red")

plot(HvC_full_model$adj.P.Val, HvC_Heart_fit_all$adj.P.Val, xlab = "Full model- human and chimp hearts", ylab = "Model with human and chimp hearts only", main = "BH-adj P value for the beta")
abline(a = 0, b = 1, col = "red")

# Get the expression values for the genes that we have average methylation data

methyl_genes_in_exp <- rownames(cpm.voom.cyclic$E) %in% methyl_values_for_exp_no_H1H$X
methyl_genes_in_exp <- as.data.frame(methyl_genes_in_exp)
summary(methyl_genes_in_exp )

counts_genes_in <- cbind(cpm.voom.cyclic$E, methyl_genes_in_exp)
counts_genes_in_cutoff <- subset(counts_genes_in, methyl_genes_in_exp == "TRUE")
exp_levels_3776 <- counts_genes_in_cutoff[,1:47]

# Make ensg names the last column

exp_levels_3776[,48] <- rownames(exp_levels_3776)

# Merge expression and methylation

exp_methyl <- merge(exp_levels_3776, methyl_values_for_exp_no_H1H, by.x = "V48", by.y = "X")
rownames(exp_methyl) <- exp_methyl$V48
dim(exp_methyl)

#write.table(exp_methyl, "../human_chimp_orth_exp_methyl_7725_hum.txt", quote = F)

# Separate expression and methylation

expression_values_only <- exp_methyl[,2:32]
methylation_values_only <- exp_methyl[,49:79]
```

```{r}
# Select data- human versus chimp heart

limma.ash <- readRDS("../data/results-limma-voom-ash.rds", refhook = NULL)
human_chimp_heart <- limma.ash$HvC_Heart

summary(human_chimp_heart$adj.P.Val < 0.01)
```


# Human v Chimp Heart

## Obtain human and chimp heart data

```{r}
## Obtain human and chimp heart data

chimp_human_heart <- c(1, 5, 9, 13, 20, 24, 28)

# First, pick out the genes that are DE (you run the entire analysis only on genes that are DE)


## Prepare to make the linear model 

species_no_rhesus <- species[chimp_human_heart]
species_no_rhesus <- droplevels.factor(species_no_rhesus, "Rhesus")

# Make the design matrix

design <- model.matrix(~ as.factor(species_no_rhesus) + RIN[chimp_human_heart,])

#write.table(cpm.voom.cyclic, "../data/human_chimp_orth_cpm_voom_cyclic.txt")

fit_all <- lmFit(cpm.voom.cyclic[,chimp_human_heart], design)
fit_all <- eBayes(fit_all)

HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust="BH", number=Inf, sort.by="none")

summary(HvC_Heart_fit_all$logFC)

summary(HvC_Heart_fit_all$adj.P.Val < .01)
summary(HvC_Heart_fit_all$adj.P.Val < .05)
summary(HvC_Heart_fit_all$adj.P.Val < .1)

# Genes DE at FDR 5% and we have methyl data for it

HvC_Heart_fit_all_5perc <- HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val < 0.05), ]

human_chimp_heart <-  rownames(exp_methyl) %in% HvC_Heart_fit_all_5perc$genes
human_chimp_heart <- as.data.frame(human_chimp_heart)
counts_genes_in <- cbind(exp_methyl, human_chimp_heart)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart == "TRUE")
counts_genes_in_cutoff <- counts_genes_in_cutoff[,1:79]  

expression_values_only <- counts_genes_in_cutoff[,2:32]
methylation_values_only <- counts_genes_in_cutoff[,49:79]

## Obtain corrected data (regression for scenario 2)

# Make an array to hold the residuals

resid_methyl <- array(0, dim = c(nrow(expression_values_only), 7)) 

# Regress out RIN on a gene-by-gene basis
expression_values <- as.data.frame(expression_values)
methylation_values <- as.data.frame(methylation_values)


for (i in 1:nrow(expression_values_only)){
    resid_methyl[i,] <- lm(t(expression_values_only[i,chimp_human_heart]) ~ t(methylation_values_only[i,chimp_human_heart]))$resid
}

rownames(resid_methyl) <- rownames(expression_values_only)

# Scenario 1

fit1 <- lmFit(expression_values_only[, chimp_human_heart], design)
fit1 <- eBayes(fit1)
HvC_Heart_fit1 = topTable(fit1, coef=2, adjust="BH", number=Inf, sort.by="none")

# Scenario 2

fit2 <- lmFit(resid_methyl, design)
fit2 <- eBayes(fit2)
HvC_Heart_fit2 = topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")

# Bind DE from fit1 and DE from fit2 together

HvC_Heart_fits12 <- as.data.frame(cbind(rownames(HvC_Heart_fit1), HvC_Heart_fit1$adj.P.Val, HvC_Heart_fit2$adj.P.Val), stringsAsFactors = FALSE)
HvC_Heart_fits12[,2] <- as.numeric(HvC_Heart_fits12[,2])
HvC_Heart_fits12[,3] <- as.numeric(HvC_Heart_fits12[,3])
colnames(HvC_Heart_fits12) <- c("genes", "fit1_FDR", "fit2_FDR")

# Set FDR level

FDR_level <- 0.05

DE_both <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_both)

DE_before <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR < FDR_level &  HvC_Heart_fits12$fit2_FDR > FDR_level),]
nrow(DE_before)

DE_after <- HvC_Heart_fits12[which(HvC_Heart_fits12$fit1_FDR > FDR_level &  HvC_Heart_fits12$fit2_FDR < FDR_level),]
nrow(DE_after)


### Permute the methylation values, then re-run

run1000 <- array(0, dim = c(3, 1000)) 

for (k in 1:1000){

# Run 1 for all genes
run1 <- array(0, dim = c(nrow(expression_values_only), 7)) 

for (i in 1:nrow(expression_values_only)){
  df <- as.data.frame(methylation_values_only[i,chimp_human_heart])
  new_df <- as.data.frame(sample(df))
  run1[i,] <- t(new_df)
}



# Make an array to hold the residuals

resid_methyl_perm <- array(0, dim = c(nrow(expression_values_only), 7)) 


#for (i in 1:2){
for (i in 1:nrow(expression_values_only)){
    resid_methyl_perm[i,] <- lm(t(expression_values_only[i,chimp_human_heart]) ~ run1[i,])$resid
}

rownames(resid_methyl_perm) <- rownames(expression_values_only)


# Scenario 2

fit2 <- lmFit(resid_methyl_perm, design)
fit2 <- eBayes(fit2)
HvC_Heart_perm = topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")


HvCHeart_DE_perm <- cbind(HvC_Heart_fits12$fit1_FDR, HvC_Heart_perm$adj.P.Val)

DE_total <- as.data.frame(HvCHeart_DE_perm[which(HvCHeart_DE_perm[,1] < FDR_level & HvCHeart_DE_perm[,2]  < FDR_level),])
DE_total_report <- nrow(DE_total)

DE_before <- as.data.frame(HvCHeart_DE_perm[which(HvCHeart_DE_perm[,1]  < FDR_level &  HvCHeart_DE_perm[,2]  > FDR_level),])
DE_before_report <- nrow(DE_before)

DE_after <- as.data.frame(HvCHeart_DE_perm[which(HvCHeart_DE_perm[,1] > FDR_level &  HvCHeart_DE_perm[,2] < FDR_level),])
DE_after_report <- nrow(DE_after)

new_numbers <- rbind(DE_total_report, DE_before_report, DE_after_report)

run1000[,k] <- new_numbers
}

median(run1000[1,1:1000])
median(run1000[2,1:1000])
median(run1000[3,1:1000])


```



```{r}
# Separate expression and methylation

expression_values_only <- exp_methyl[,2:32]
methylation_values_only <- exp_methyl[,49:79]
```

