---
title: "Methylation_probes"
author: "Lauren Blake"
date: "June 12, 2017"
output: pdf_document
---

This is a script to explore the methylation levels upsteam of the TSS. 

```{r}
# Set length from TSS

TSS_length <- 20000

# Load needed libraries

library(plyr)

# Get hg19 TSS data (from https://sourceforge.net/projects/seqminer/files/Reference%20coordinate/)
refGene_hg19_TSS <- read.delim("../../../Downloads/refGene_hg19_TSS.bed", header=FALSE)
refGene_hg19_TSS <- refGene_hg19_TSS[,-4]
refGene_hg19_TSS[,1] <- as.character(refGene_hg19_TSS[,1])
refGene_hg19_TSS[,4] <- as.character(refGene_hg19_TSS[,4])
refGene_hg19_TSS[,5] <- as.character(refGene_hg19_TSS[,5])

length(unique(refGene_hg19_TSS[,4]))
```


```{r}
#Open the orthologous exons file

metaOrthoExon <- read.delim("~/Desktop/Reg_Evo_Primates/data/metaOrthoExonTrios.0.92.0.96.wExonEnsID.wSymbols.txt", header=FALSE)

# Now, I want to remake this but with the value of exon #1

uniq_plus <- c(2,3,4,5,7,11,15,16)

# Make data frame
non_uniq_ENSG_gene_names <- as.data.frame(unique(metaOrthoExon[,uniq_plus], incomparables = FALSE))

# Subset so you only have ortho exon #1 

uniq_ENSG_gene_names <- non_uniq_ENSG_gene_names[which(non_uniq_ENSG_gene_names$V3 == 1),]
uniq_ENSG_gene_names <- uniq_ENSG_gene_names[,-2]
uniq_ENSG_gene_names[,1] <- as.character(uniq_ENSG_gene_names[,1])
uniq_ENSG_gene_names[,2] <- as.character(uniq_ENSG_gene_names[,2])
uniq_ENSG_gene_names[,3] <- as.integer(uniq_ENSG_gene_names[,3])
uniq_ENSG_gene_names[,5] <- as.character(uniq_ENSG_gene_names[,5])
uniq_ENSG_gene_names[,6] <- as.character(uniq_ENSG_gene_names[,6])
uniq_ENSG_gene_names[,7] <- as.character(uniq_ENSG_gene_names[,7])

# Replace -1 with (-) and 1 with (+)

uniq_ENSG_gene_names$V7 <- gsub("-1", "-", uniq_ENSG_gene_names$V7)
uniq_ENSG_gene_names$V7 <- gsub("1", "+", uniq_ENSG_gene_names$V7)

# Make sure that there are 30,030 metaOrthoExons
dim(uniq_ENSG_gene_names)
str(uniq_ENSG_gene_names)

# Label the columns

colnames(uniq_ENSG_gene_names) <- c("ENSG", "chromosome", "orth_exon_1", "strand", "strand_chimp", "strand_rhesus", "gene_name")

# Import the names of genes passing CPM threshold 

cpm_12184 <- read.delim("../data/cpm_12184.txt")

# Sort the orthologous exon genes file so that we are only working with the names of the orthologous genes from the RNA seq data

inshared_lists = uniq_ENSG_gene_names$ENSG %in% rownames(cpm_12184)
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(uniq_ENSG_gene_names, inshared_lists_data)
counts_genes_in_ortho <- subset(counts_genes_in, inshared_lists_data == "TRUE")
counts_genes_in_ortho <- counts_genes_in_ortho[,1:7]

# Note: counts_genes_in_ortho are the 12,184 genes from the RNA seq data

# Sort the hg_TSS bed file so that we are only working with the names of the orthologous exon genes

inshared_lists = refGene_hg19_TSS$V5 %in% counts_genes_in_ortho$gene_name
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(refGene_hg19_TSS, inshared_lists_data)
counts_genes_in_TSS <- subset(counts_genes_in, inshared_lists_data == "TRUE")
counts_genes_in_TSS <- counts_genes_in_TSS[,1:5]

colnames(counts_genes_in_TSS) <- c("chromosome", "TSS_start_pos", "TSS_start_neg", "gene_name", "strand")

total <- merge(counts_genes_in_TSS, counts_genes_in_ortho, by=c("chromosome","gene_name","strand"))

dim(total)

# Many of these 22576 entries are driven by multiple TSS for the same gene 

# Find the samples with only one ENSG entry

ENSG_freq_table <- count(total, 'ENSG')
solo_ENSG_freq_table <- ENSG_freq_table[which(ENSG_freq_table[,2] == 1), ]
nrow(solo_ENSG_freq_table)

# 5907/11128 genes are not duplicated 

# These are the ones that are not duplicated -- good to go

inshared_lists = total$ENSG %in% solo_ENSG_freq_table[,1]
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(total, inshared_lists_data)
counts_genes_in_TSS1 <- subset(counts_genes_in, inshared_lists_data == "TRUE")
counts_genes_in_TSS1 <- counts_genes_in_TSS1[,1:9]
dim(counts_genes_in_TSS1)

# These are the ones that are duplicates -- take the TSS closest to the first 

multiple_ENSG_freq_table <- ENSG_freq_table[which(ENSG_freq_table[,2] != 1), ]
nrow(multiple_ENSG_freq_table)

# 5221/11128 genes are duplicated 

inshared_lists = total$ENSG %in% multiple_ENSG_freq_table$ENSG
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(total, inshared_lists_data)
counts_genes_in_TSS2 <- subset(counts_genes_in, inshared_lists_data == "TRUE")
counts_genes_in_TSS2 <- counts_genes_in_TSS2[,1:9]

dim(counts_genes_in_TSS2)

# Collapse same entries

multiple_TSS <- unique(counts_genes_in_TSS2[,1:9])

# Take the single entries -- also good to go

ENSG_freq_table2 <- count(multiple_TSS, 'ENSG')
solo_ENSG_freq_table2 <- ENSG_freq_table2[which(ENSG_freq_table2[,2] == 1), ]
nrow(solo_ENSG_freq_table2)

inshared_lists = multiple_TSS$ENSG %in% solo_ENSG_freq_table2[,1]
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(multiple_TSS, inshared_lists_data)
counts_genes_in_TSS3 <- subset(counts_genes_in, inshared_lists_data == "TRUE")
counts_genes_in_TSS3 <- counts_genes_in_TSS3[,1:9]
dim(counts_genes_in_TSS3)

# We have retained another 3027 genes

multiple_ENSG_freq_table2 <- ENSG_freq_table2[which(ENSG_freq_table2[,2] != 1), ]
nrow(multiple_ENSG_freq_table2)

# Find the distance from TSS to the human orthologous exon 1

inshared_lists = multiple_TSS$ENSG %in% multiple_ENSG_freq_table2[,1]
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(multiple_TSS, inshared_lists_data)
counts_genes_in_TSS4 <- subset(counts_genes_in, inshared_lists_data == "TRUE")
counts_genes_in_TSS4 <- counts_genes_in_TSS4[,1:9]


diff_TSS_orth_exon <- abs(counts_genes_in_TSS4$TSS_start_pos - counts_genes_in_TSS4$orth_exon_1)

multiple_TSS_dist <- cbind(counts_genes_in_TSS4, diff_TSS_orth_exon)

get_min_set_TSS <- multiple_TSS_dist[as.logical(ave(multiple_TSS_dist$diff_TSS_orth_exon, multiple_TSS_dist$ENSG, FUN = function(x) x == min(x))),]
get_min_set_TSS <- get_min_set_TSS[,1:9]
dim(get_min_set_TSS)

# Now, we have 3 sets that we want

# 5907 from counts_genes_in_TSS1
# 3027 from counts_genes_in_TSS3
# 2196 from get_min_set_TSS 

info_orth_genes <- rbind(counts_genes_in_TSS1, counts_genes_in_TSS3, get_min_set_TSS)
dim(info_orth_genes)

info_orth_genes$strand <- as.numeric(as.factor(info_orth_genes$strand))

# Replace (-) with 1 and (+) with 2 and subset by 1 and 2, then perform the two functions, 1 on each of the subsets

info_orth_genes_neg <- info_orth_genes[which(info_orth_genes$strand == 1), ]
dim(info_orth_genes_neg) 

info_orth_genes_neg[,10] <- info_orth_genes_neg$TSS_start_neg + TSS_length

info_orth_genes_pos <- info_orth_genes[which(info_orth_genes$strand == 2), ]
dim(info_orth_genes_pos)

info_orth_genes_pos[,10] <- info_orth_genes_pos$TSS_start_pos - TSS_length

#info_orth_genes_dist <- rbind(info_orth_genes_pos, info_orth_genes_neg)
#dim(info_orth_genes_dist)
```

### Find which orthologous CpGs are within X bp of the TSS

```{r}

# For positive stranded genes

none_fitting <- matrix(c(NA, NA, NA, NA, NA, NA,NA, NA, NA, NA), nrow =1, ncol = 10)

check_pos_strand <- function(chrom_to_match, cpg_coor_to_test){
    pos_in_interval <- info_orth_genes_pos[ (info_orth_genes_pos$chromosome == chrom_to_match) & (info_orth_genes_pos$V10 <= cpg_coor_to_test) & (cpg_coor_to_test <= info_orth_genes_pos$TSS_start_pos),   ]
    if (nrow(pos_in_interval) == 0) none_fitting else pos_in_interval
} 

# For negative stranded genes

check_neg_strand <- function(chrom_to_match, cpg_coor_to_test){
  neg_in_interval <- info_orth_genes_neg[ (info_orth_genes_neg$chromosome == chrom_to_match) & (info_orth_genes_neg$V10 >= cpg_coor_to_test) & (cpg_coor_to_test >= info_orth_genes_neg$TSS_start_neg),   ]
  if (nrow(neg_in_interval) == 0) none_fitting else neg_in_interval
} 

#OneNum <- lapply(OneNum$MiddleNum, check_pos_strand)

# Make toy example

#test_matrix <- as.data.frame(matrix(c("chr1", "chr1", "chr2", 2, 236849759, 1243270), nrow = 3, ncol = 2))

#test_df1 <- as.data.frame(matrix(c("chr1", "chr1", "chr2"), nrow = 3, ncol = 1))
#test_df2 <- as.data.frame(c(2, 236849759, 1243270))
#test_df <- cbind(test_df1, test_df2) 

#test_df[,1] <- as.character(test_df[,1])
#test_df[,2] <- as.numeric(test_df[,2])
#colnames(test_df) <- c("chromosome", "coor")
#str(test_df)

# Run toy example
#test_coor <- lapply(test_df$coor, check_pos_strand)

# Make results into a data frame
#df_test_coor_pos <- data.frame(matrix(unlist(test_coor), nrow=nrow(test_df), byrow=T),stringsAsFactors=FALSE)

# Check chromosomes are congruent within a row
#chrom_df_test_coor_pos <- cbind(test_df$chromosome, test_df$coor, df_test_coor_pos)

# If chromosome #s match, then keep
#names_ENSG_pos <- chrom_df_test_coor_pos[which(chrom_df_test_coor_pos[,1] == chrom_df_test_coor_pos[,3]), ]

# Import orthologous CpG sites w/ sufficient coverage

#orth_methyl_chrom_cpg <- read.delim("../data/orth_methyl_chrom_cpg.txt", stringsAsFactors = FALSE)

read_in <- read.delim("test_output.txt", header = TRUE, stringsAsFactors = FALSE, as.is = TRUE)

sep_methyl <- data.frame(do.call('rbind', strsplit(as.character(read_in$C1H),':',fixed=TRUE)))
sep_methyl[,1] <- as.character(sep_methyl[,1])
sep_methyl[,2] <- as.integer(as.character(sep_methyl[,2]))
dim(sep_methyl)
str(sep_methyl)
colnames(sep_methyl) <- c("chromosome", "cpg_position")

test_orth_methyl_chrom_cpg <- sep_methyl[1:100,]
#sep_methyl[1945,]
#check_pos_strand("chr1", 1369132)

#newdata <- info_orth_genes[order(info_orth_genes$TSS_start_pos),]

# Run example with pos strand 
#test_coor <- lapply(test_orth_methyl_chrom_cpg$cpg_position, check_pos_strand)

# Instead of lapply, we want to be able to feed it chromosome and position

# Make a place to store results
pos_results <- data.frame(matrix(NA, nrow = nrow(test_orth_methyl_chrom_cpg), ncol = 10))

# Run check_pos_strand on all orthologous CpGs

# Joyce- doesn't work correctly (see lines 135-142)

i = 1
for (i in 1:nrow(test_orth_methyl_chrom_cpg)){
  chrom_to_match <- test_orth_methyl_chrom_cpg$chromosome[i]
  cpg_coor_to_test <- test_orth_methyl_chrom_cpg$cpg_position[i]
  pos_results[i, ] <- check_pos_strand(chrom_to_match, cpg_coor_to_test)
  
  print(i)
}

#pos_results
#for (i in 1:nrow(test_orth_methyl_chrom_cpg)){
#  chrom_to_match <- test_orth_methyl_chrom_cpg$chromosome[i]
#  cpg_coor_to_test <- test_orth_methyl_chrom_cpg$cpg_position[i]
#  pos_results[i,] <- as.data.frame(check_pos_strand(chrom_to_match, cpg_coor_to_test))
#  print(i)
#}
  

# Make results into a data frame
#df_test_coor_pos <- data.frame(matrix(unlist(pos_results), nrow=nrow(test_orth_methyl_chrom_cpg), byrow=T),stringsAsFactors=FALSE)

# Check chromosomes are congruent within a row
#chrom_df_test_coor_pos <- cbind(test_orth_methyl_chrom_cpg$chromosome,test_orth_methyl_chrom_cpg$cpg_position, df_test_coor_pos)

# If chromosome #s match, then keep
#names_ENSG_pos <- chrom_df_test_coor_pos[which(chrom_df_test_coor_pos[,1] == chrom_df_test_coor_pos[,3]), ]
#dim(names_ENSG_pos)

#check_neg_strand("chr1", 1243269)
#chrom_to_match <- "chr1"
#cpg_coor_to_test <- 1231505

#as.data.frame(check_pos_strand("chr1", 2162774))


# Make a place to store results
neg_results <- data.frame(matrix(NA, nrow = nrow(test_orth_methyl_chrom_cpg), ncol = 10))

# Run check_pos_strand on all orthologous CpGs

i = 1
for (i in 1:nrow(test_orth_methyl_chrom_cpg)){
  chrom_to_match <- test_orth_methyl_chrom_cpg$chromosome[i]
  cpg_coor_to_test <- test_orth_methyl_chrom_cpg$cpg_position[i]
  neg_results[i, ] <- data.frame(check_neg_strand(chrom_to_match, cpg_coor_to_test))
  
  print(i)
}

```

```{r}
# Joyce- works correctly but simply prints rather than stores the results
#i = 1
#for (i in 1:nrow(test_orth_methyl_chrom_cpg)){
#  chrom_to_match <- test_orth_methyl_chrom_cpg$chromosome[i]
#  cpg_coor_to_test <- test_orth_methyl_chrom_cpg$cpg_position[i]
#  print(i)
#  print(check_pos_strand(chrom_to_match, cpg_coor_to_test))
  
  #print(i)
#}
```



